// @ui5/webcomponents-ngx/esm2022/main/multi-input/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/MultiInput.js';
import * as i0 from "@angular/core";
import * as i1 from "@ui5/webcomponents-ngx/generic-cva";
let MultiInputComponent = class MultiInputComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiInputComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MultiInputComponent, isStandalone: true, selector: "ui5-multi-input", inputs: { disabled: ["disabled", "disabled", booleanAttribute], placeholder: "placeholder", readonly: ["readonly", "readonly", booleanAttribute], required: ["required", "required", booleanAttribute], noTypeahead: ["noTypeahead", "noTypeahead", booleanAttribute], type: "type", value: "value", valueState: "valueState", name: "name", showSuggestions: ["showSuggestions", "showSuggestions", booleanAttribute], maxlength: "maxlength", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", showClearIcon: ["showClearIcon", "showClearIcon", booleanAttribute], open: ["open", "open", booleanAttribute], showValueHelpIcon: ["showValueHelpIcon", "showValueHelpIcon", booleanAttribute] }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input", ui5SelectionChange: "ui5SelectionChange", ui5Open: "ui5Open", ui5Close: "ui5Close", ui5ValueHelpTrigger: "ui5ValueHelpTrigger", ui5TokenDelete: "ui5TokenDelete" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5MultiInput"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MultiInputComponent = __decorate([
    ProxyInputs([
        'disabled',
        'placeholder',
        'readonly',
        'required',
        'noTypeahead',
        'type',
        'value',
        'valueState',
        'name',
        'showSuggestions',
        'maxlength',
        'accessibleName',
        'accessibleNameRef',
        'showClearIcon',
        'open',
        'showValueHelpIcon',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'input: ui5Input',
        'selection-change: ui5SelectionChange',
        'open: ui5Open',
        'close: ui5Close',
        'value-help-trigger: ui5ValueHelpTrigger',
        'token-delete: ui5TokenDelete',
    ]),
    __metadata("design:paramtypes", [])
], MultiInputComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiInputComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-multi-input',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'placeholder',
                        'readonly',
                        'required',
                        'noTypeahead',
                        'type',
                        'value',
                        'valueState',
                        'name',
                        'showSuggestions',
                        'maxlength',
                        'accessibleName',
                        'accessibleNameRef',
                        'showClearIcon',
                        'open',
                        'showValueHelpIcon',
                    ],
                    outputs: [
                        'ui5Change',
                        'ui5Input',
                        'ui5SelectionChange',
                        'ui5Open',
                        'ui5Close',
                        'ui5ValueHelpTrigger',
                        'ui5TokenDelete',
                    ],
                    exportAs: 'ui5MultiInput',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], noTypeahead: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showSuggestions: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showClearIcon: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showValueHelpIcon: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { MultiInputComponent };
