// @ui5/webcomponents-ngx/esm2022/main/panel/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Panel.js';
import * as i0 from "@angular/core";
let PanelComponent = class PanelComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PanelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: PanelComponent, isStandalone: true, selector: "ui5-panel", inputs: { headerText: "headerText", fixed: ["fixed", "fixed", booleanAttribute], collapsed: ["collapsed", "collapsed", booleanAttribute], noAnimation: ["noAnimation", "noAnimation", booleanAttribute], accessibleRole: "accessibleRole", headerLevel: "headerLevel", accessibleName: "accessibleName", stickyHeader: ["stickyHeader", "stickyHeader", booleanAttribute] }, outputs: { ui5Toggle: "ui5Toggle" }, exportAs: ["ui5Panel"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
PanelComponent = __decorate([
    ProxyInputs([
        'headerText',
        'fixed',
        'collapsed',
        'noAnimation',
        'accessibleRole',
        'headerLevel',
        'accessibleName',
        'stickyHeader',
    ]),
    ProxyOutputs(['toggle: ui5Toggle']),
    __metadata("design:paramtypes", [])
], PanelComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PanelComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-panel',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'headerText',
                        'fixed',
                        'collapsed',
                        'noAnimation',
                        'accessibleRole',
                        'headerLevel',
                        'accessibleName',
                        'stickyHeader',
                    ],
                    outputs: ['ui5Toggle'],
                    exportAs: 'ui5Panel',
                }]
        }], ctorParameters: () => [], propDecorators: { fixed: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], collapsed: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], noAnimation: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], stickyHeader: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { PanelComponent };
