// @ui5/webcomponents-ngx/esm2022/main/split-button/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SplitButton.js';
import * as i0 from "@angular/core";
let SplitButtonComponent = class SplitButtonComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SplitButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SplitButtonComponent, isStandalone: true, selector: "ui5-split-button", inputs: { icon: "icon", activeArrowButton: ["activeArrowButton", "activeArrowButton", booleanAttribute], design: "design", disabled: ["disabled", "disabled", booleanAttribute], accessibleName: "accessibleName" }, outputs: { ui5Click: "ui5Click", ui5ArrowClick: "ui5ArrowClick" }, exportAs: ["ui5SplitButton"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SplitButtonComponent = __decorate([
    ProxyInputs([
        'icon',
        'activeArrowButton',
        'design',
        'disabled',
        'accessibleName',
    ]),
    ProxyOutputs(['click: ui5Click', 'arrow-click: ui5ArrowClick']),
    __metadata("design:paramtypes", [])
], SplitButtonComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SplitButtonComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-split-button',
                    template: '<ng-content></ng-content>',
                    inputs: ['icon', 'activeArrowButton', 'design', 'disabled', 'accessibleName'],
                    outputs: ['ui5Click', 'ui5ArrowClick'],
                    exportAs: 'ui5SplitButton',
                }]
        }], ctorParameters: () => [], propDecorators: { activeArrowButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { SplitButtonComponent };
