// @ui5/webcomponents-ngx/esm2022/main/table/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Table.js';
import * as i0 from "@angular/core";
let TableComponent = class TableComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TableComponent, isStandalone: true, selector: "ui5-table", inputs: { accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", noDataText: "noDataText", overflowMode: "overflowMode", loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay" }, outputs: { ui5RowClick: "ui5RowClick" }, exportAs: ["ui5Table"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableComponent = __decorate([
    ProxyInputs([
        'accessibleName',
        'accessibleNameRef',
        'noDataText',
        'overflowMode',
        'loading',
        'loadingDelay',
    ]),
    ProxyOutputs(['row-click: ui5RowClick']),
    __metadata("design:paramtypes", [])
], TableComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accessibleName',
                        'accessibleNameRef',
                        'noDataText',
                        'overflowMode',
                        'loading',
                        'loadingDelay',
                    ],
                    outputs: ['ui5RowClick'],
                    exportAs: 'ui5Table',
                }]
        }], ctorParameters: () => [], propDecorators: { loading: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { TableComponent };
