// @ui5/webcomponents-ngx/esm2022/main/multi-combo-box/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/MultiComboBox.js';
import * as i0 from "@angular/core";
import * as i1 from "@ui5/webcomponents-ngx/generic-cva";
let MultiComboBoxComponent = class MultiComboBoxComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiComboBoxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MultiComboBoxComponent, isStandalone: true, selector: "ui5-multi-combobox", inputs: { value: "value", name: "name", noTypeahead: ["noTypeahead", "noTypeahead", booleanAttribute], placeholder: "placeholder", noValidation: ["noValidation", "noValidation", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], valueState: "valueState", readonly: ["readonly", "readonly", booleanAttribute], required: ["required", "required", booleanAttribute], filter: "filter", showClearIcon: ["showClearIcon", "showClearIcon", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", showSelectAll: ["showSelectAll", "showSelectAll", booleanAttribute] }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input", ui5Open: "ui5Open", ui5Close: "ui5Close", ui5SelectionChange: "ui5SelectionChange" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5MultiCombobox"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MultiComboBoxComponent = __decorate([
    ProxyInputs([
        'value',
        'name',
        'noTypeahead',
        'placeholder',
        'noValidation',
        'disabled',
        'valueState',
        'readonly',
        'required',
        'filter',
        'showClearIcon',
        'accessibleName',
        'accessibleNameRef',
        'showSelectAll',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'input: ui5Input',
        'open: ui5Open',
        'close: ui5Close',
        'selection-change: ui5SelectionChange',
    ]),
    __metadata("design:paramtypes", [])
], MultiComboBoxComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiComboBoxComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-multi-combobox',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'value',
                        'name',
                        'noTypeahead',
                        'placeholder',
                        'noValidation',
                        'disabled',
                        'valueState',
                        'readonly',
                        'required',
                        'filter',
                        'showClearIcon',
                        'accessibleName',
                        'accessibleNameRef',
                        'showSelectAll',
                    ],
                    outputs: [
                        'ui5Change',
                        'ui5Input',
                        'ui5Open',
                        'ui5Close',
                        'ui5SelectionChange',
                    ],
                    exportAs: 'ui5MultiCombobox',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { noTypeahead: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], noValidation: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showClearIcon: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showSelectAll: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { MultiComboBoxComponent };
