// @ui5/webcomponents-ngx/esm2022/main/toolbar-separator/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarSeparator.js';
import * as i0 from "@angular/core";
let ToolbarSeparatorComponent = class ToolbarSeparatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSeparatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarSeparatorComponent, isStandalone: true, selector: "ui5-toolbar-separator", inputs: { overflowPriority: "overflowPriority", preventOverflowClosing: ["preventOverflowClosing", "preventOverflowClosing", booleanAttribute] }, exportAs: ["ui5ToolbarSeparator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarSeparatorComponent = __decorate([
    ProxyInputs(['overflowPriority', 'preventOverflowClosing']),
    __metadata("design:paramtypes", [])
], ToolbarSeparatorComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSeparatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-separator',
                    template: '<ng-content></ng-content>',
                    inputs: ['overflowPriority', 'preventOverflowClosing'],
                    exportAs: 'ui5ToolbarSeparator',
                }]
        }], ctorParameters: () => [], propDecorators: { preventOverflowClosing: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ToolbarSeparatorComponent };
