// @ui5/webcomponents-ngx/esm2022/main/tab-separator/index.mjs
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import '@ui5/webcomponents/dist/TabSeparator.js';
import * as i0 from "@angular/core";
class TabSeparatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabSeparatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TabSeparatorComponent, isStandalone: true, selector: "ui5-tab-separator", exportAs: ["ui5TabSeparator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabSeparatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tab-separator',
                    template: '<ng-content></ng-content>',
                    exportAs: 'ui5TabSeparator',
                }]
        }], ctorParameters: () => [] });
export { TabSeparatorComponent };
