// @ui5/webcomponents-react/wrappers/themes/CustomVariables.js
export let CustomVariables = /*#__PURE__*/function (CustomVariables) {
  CustomVariables["ActionSheetBoxShadow"] = "--_ui5wcr_ActionSheet_BoxShadow";
  CustomVariables["ActionSheetHeaderBoxShadow"] = "--_ui5wcr_ActionSheet_HeaderBoxShadow";
  CustomVariables["ActionSheetMobileHeaderBackground"] = "--_ui5wcr_ActionSheet_MobileHeaderBackground";
  CustomVariables["ActionSheetMobileHeaderBorderRadius"] = "--_ui5wcr_ActionSheet_MobileHeaderBorderRadius";
  CustomVariables["ActionSheetMobileHeaderBoxShadow"] = "--_ui5wcr_ActionSheet_MobileHeaderBoxShadow";
  CustomVariables["ActionSheetMobileHeaderTextColor"] = "--_ui5wcr_ActionSheet_MobileHeaderTextColor";
  CustomVariables["ActionSheetTopBorderRadius"] = "--_ui5wcr_ActionSheet_TopBorderRadius";
  CustomVariables["AnalyticalTableHeaderActiveTextColor"] = "--_ui5wcr-AnalyticalTable-HeaderActiveTextColor";
  CustomVariables["AnalyticalTableHeaderBorderWidth"] = "--_ui5wcr-AnalyticalTable-HeaderBorderWidth";
  CustomVariables["AnalyticalTableHeaderFontFamily"] = "--_ui5wcr-AnalyticalTable-HeaderFontFamily";
  CustomVariables["AnalyticalTableOuterBorderBlock"] = "--_ui5wcr-AnalyticalTable-OuterBorderBlock";
  CustomVariables["AnalyticalTableOuterBorderInline"] = "--_ui5wcr-AnalyticalTable-OuterBorderInline";
  CustomVariables["AnalyticalTableOuterCellBorder"] = "--_ui5wcr-AnalyticalTable-OuterCellBorder";
  CustomVariables["FormHeaderBorderWidth"] = "--_ui5wcr_Form_HeaderBorderWidth";
  CustomVariables["ObjectPageSectionBorderTop"] = "--_ui5wcr_ObjectPage_SectionBorderTop";
  CustomVariables["ObjectPageSectionTitleFontFamily"] = "--_ui5wcr_ObjectPage_SectionTitleFontFamily";
  CustomVariables["ObjectPageSectionTitleHeight"] = "--_ui5wcr_ObjectPage_SectionTitleHeight";
  CustomVariables["ObjectPageSectionTitleLineHeight"] = "--_ui5wcr_ObjectPage_SectionTitleLineHeight";
  CustomVariables["ObjectPageSectionTitleTextDecoration"] = "--_ui5wcr_ObjectPage_SectionTitleTextDecoration";
  CustomVariables["ObjectPageSubSectionBackgroundColor"] = "--_ui5wcr_ObjectPage_SubSectionBackgroundColor";
  CustomVariables["ObjectPageSubSectionBorderRadius"] = "--_ui5wcr_ObjectPage_SubSectionBorderRadius";
  CustomVariables["ObjectPageSubSectionPaddingBlockStart"] = "--_ui5wcr_ObjectPage_PaddingBlockStart";
  CustomVariables["SplitterBarBorderFix"] = "--_ui5wcr_Splitter_BarBorderHighContrastFix";
  CustomVariables["SplitterBarBorderFocus"] = "--_ui5wcr_Splitter_BarBorderFocus";
  CustomVariables["SplitterBarBorderStyle"] = "--_ui5wcr_Splitter_BarBorderStyle";
  CustomVariables["SplitterBarOutline"] = "--_ui5wcr_Splitter_BarOutline";
  CustomVariables["SplitterContentBorderColor"] = "--_ui5wcr_Splitter_ContentBorderColor";
  CustomVariables["SplitterIconColor"] = "--_ui5wcr_Splitter_IconColor";
  CustomVariables["ToolbarFocusOutline"] = "--_ui5wcr_Toolbar_FocusOutline";
  CustomVariables["ToolbarFocusShadow"] = "--_ui5wcr_Toolbar_FocusShadow";
  CustomVariables["ScrollbarBorder"] = "--_ui5wcr_Scrollbar_Border";
  CustomVariables["ScrollbarBorderRadius"] = "--_ui5wcr_Scrollbar_BorderRadius";
  return CustomVariables;
}({});
export const CustomThemingParameters = new Proxy(CustomVariables, {
  get: (target, prop) => `var(${target[prop]})`
});
