// @ui5/webcomponents-ngx/esm2022/main/check-box/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CheckBox.js';
import * as i0 from "@angular/core";
import * as i1 from "@ui5/webcomponents-ngx/generic-cva";
let CheckBoxComponent = class CheckBoxComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.checked = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.checked;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CheckBoxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: CheckBoxComponent, isStandalone: true, selector: "ui5-checkbox", inputs: { accessibleNameRef: "accessibleNameRef", accessibleName: "accessibleName", disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], displayOnly: ["displayOnly", "displayOnly", booleanAttribute], required: ["required", "required", booleanAttribute], indeterminate: ["indeterminate", "indeterminate", booleanAttribute], checked: ["checked", "checked", booleanAttribute], text: "text", valueState: "valueState", wrappingType: "wrappingType", name: "name" }, outputs: { ui5Change: "ui5Change" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5Checkbox"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CheckBoxComponent = __decorate([
    ProxyInputs([
        'accessibleNameRef',
        'accessibleName',
        'disabled',
        'readonly',
        'displayOnly',
        'required',
        'indeterminate',
        'checked',
        'text',
        'valueState',
        'wrappingType',
        'name',
    ]),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], CheckBoxComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CheckBoxComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-checkbox',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accessibleNameRef',
                        'accessibleName',
                        'disabled',
                        'readonly',
                        'displayOnly',
                        'required',
                        'indeterminate',
                        'checked',
                        'text',
                        'valueState',
                        'wrappingType',
                        'name',
                    ],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5Checkbox',
                    hostDirectives: [GenericControlValueAccessor],
                    host: { '(change)': '_cva?.onChange?.(cvaValue);' },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], displayOnly: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], indeterminate: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], checked: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { CheckBoxComponent };
