// @ui5/webcomponents-react/dist/components/ObjectPageTitle/index.js
'use client';

import { debounce, Device, useStylesheet, useSyncRef } from '@ui5/webcomponents-react-base';
import { clsx } from 'clsx';
import { Children, cloneElement, forwardRef, isValidElement, useCallback, useEffect, useRef, useState } from 'react';
import { FlexBoxAlignItems, FlexBoxJustifyContent } from '../../enums/index.js';
import { stopPropagation } from '../../internal/stopPropagation.js';
import { flattenFragments } from '../../internal/utils.js';
import { FlexBox } from '../FlexBox/index.js';
import { Toolbar } from '../Toolbar/index.js';
import { ToolbarSeparator } from '../ToolbarSeparator/index.js';
import { ActionsSpacer } from './ActionsSpacer.js';
import { classNames, styleData } from './ObjectPageTitle.module.css.js';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
const enhanceActionsWithClick = (actions, ref) => flattenFragments(actions, Infinity).map(action => {
  if ( /*#__PURE__*/isValidElement(action)) {
    return /*#__PURE__*/cloneElement(action, {
      onClick: e => {
        if (typeof action.props?.onClick === 'function') {
          action.props.onClick(e);
        }
        // @ts-expect-error: will be replaced
        if (ref.current?.isOpen() && !e.defaultPrevented) {
          // @ts-expect-error: will be replaced
          ref.current.close();
        }
      }
    });
  }
});

/**
 * The `ObjectPageTitle` component is used to serve as title of the `ObjectPage`.
 * It can contain Breadcrumbs, Title, Subtitle, Content, KPIs and Actions.
 *
 * __Note:__ When used inside a custom component, it's essential to pass through all props, as otherwise the component won't function as intended!
 */
const ObjectPageTitle = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    actions,
    breadcrumbs,
    children,
    header,
    subHeader,
    navigationActions,
    className,
    style,
    onToggleHeaderContentVisibility,
    actionsToolbarProps,
    navigationActionsToolbarProps,
    expandedContent,
    snappedContent,
    ...rest
  } = props;
  useStylesheet(styleData, ObjectPageTitle.displayName);
  const [componentRef, dynamicPageTitleRef] = useSyncRef(ref);
  const [showNavigationInTopArea, setShowNavigationInTopArea] = useState(undefined);
  const isMounted = useRef(false);
  const [isPhone, setIsPhone] = useState(Device.getCurrentRange(dynamicPageTitleRef.current?.getBoundingClientRect().width)?.name === 'Phone');
  const containerClasses = clsx(classNames.container, isPhone && classNames.phone, className);
  const [actionsOverflowRef, syncedActionsOverflowRef] = useSyncRef(actionsToolbarProps?.overflowPopoverRef ?? null);
  const [navActionsOverflowRef, syncedNavActionsOverflowRef] = useSyncRef(navigationActionsToolbarProps?.overflowPopoverRef ?? null);
  useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  }, [isMounted]);
  const {
    onClick: _0,
    ...propsWithoutOmitted
  } = rest;
  const onHeaderClick = useCallback(e => {
    if (typeof props?.onClick === 'function') {
      props.onClick(e);
    }
    if (typeof onToggleHeaderContentVisibility === 'function' && !props?.['data-not-clickable']) {
      onToggleHeaderContentVisibility(e);
    }
  }, [props?.onClick, onToggleHeaderContentVisibility, props?.['data-not-clickable']]);
  useEffect(() => {
    const debouncedObserverFn = debounce(([titleContainer]) => {
      // Firefox implements `borderBoxSize` as a single content rect, rather than an array
      const borderBoxSize = Array.isArray(titleContainer.borderBoxSize) ? titleContainer.borderBoxSize[0] : titleContainer.borderBoxSize;
      // Safari doesn't implement `borderBoxSize`
      const titleContainerWidth = borderBoxSize?.inlineSize ?? titleContainer.target.getBoundingClientRect().width;
      setIsPhone(Device.getCurrentRange(titleContainerWidth)?.name === 'Phone');
      if (titleContainerWidth < 1280 && !showNavigationInTopArea === true && isMounted.current) {
        setShowNavigationInTopArea(true);
      } else if (titleContainerWidth >= 1280 && !showNavigationInTopArea === false && isMounted.current) {
        setShowNavigationInTopArea(false);
      }
    }, 150);
    const observer = new ResizeObserver(debouncedObserverFn);
    if (dynamicPageTitleRef.current) {
      observer.observe(dynamicPageTitleRef.current);
    }
    return () => {
      debouncedObserverFn.cancel();
      observer.disconnect();
    };
  }, [dynamicPageTitleRef.current, showNavigationInTopArea, isMounted]);
  const handleActionsToolbarClick = e => {
    stopPropagation(e);
    if (typeof actionsToolbarProps?.onClick === 'function') {
      actionsToolbarProps.onClick(e);
    }
  };
  const handleNavigationActionsToolbarClick = e => {
    stopPropagation(e);
    if (typeof navigationActionsToolbarProps?.onClick === 'function') {
      navigationActionsToolbarProps.onClick(e);
    }
  };
  return /*#__PURE__*/_jsxs(FlexBox, {
    className: containerClasses,
    style: style,
    ref: componentRef,
    "data-component-name": "ObjectPageTitle",
    onClick: onHeaderClick,
    ...propsWithoutOmitted,
    children: [(breadcrumbs || navigationActions && showNavigationInTopArea) && /*#__PURE__*/_jsxs(FlexBox, {
      justifyContent: FlexBoxJustifyContent.SpaceBetween,
      "data-component-name": "ObjectPageTitleBreadcrumbs",
      children: [breadcrumbs && /*#__PURE__*/_jsx("div", {
        className: classNames.breadcrumbs,
        onClick: stopPropagation,
        children: breadcrumbs
      }), navigationActions && showNavigationInTopArea && /*#__PURE__*/_jsxs(Toolbar, {
        tabIndex: undefined,
        role: undefined,
        ...navigationActionsToolbarProps,
        overflowButton: navigationActionsToolbarProps?.overflowButton,
        className: clsx(classNames.toolbar, navigationActionsToolbarProps?.className),
        onClick: handleNavigationActionsToolbarClick,
        "data-component-name": "ObjectPageTitleNavActions",
        onOverflowChange: navigationActionsToolbarProps?.onOverflowChange,
        overflowPopoverRef: navActionsOverflowRef,
        design: "Auto",
        toolbarStyle: "Clear",
        active: true,
        children: [/*#__PURE__*/_jsx(ActionsSpacer, {
          onClick: onHeaderClick,
          noHover: props?.['data-not-clickable']
        }), enhanceActionsWithClick(navigationActions, syncedNavActionsOverflowRef)]
      })]
    }), /*#__PURE__*/_jsxs(FlexBox, {
      alignItems: FlexBoxAlignItems.Center,
      className: classNames.middleSection,
      "data-component-name": "ObjectPageTitleMiddleSection",
      children: [/*#__PURE__*/_jsxs(FlexBox, {
        className: classNames.titleMainSection,
        children: [header && /*#__PURE__*/_jsx("div", {
          className: classNames.title,
          "data-component-name": "ObjectPageTitleHeader",
          children: header
        }), children && /*#__PURE__*/_jsx("div", {
          className: classNames.content,
          "data-component-name": "ObjectPageTitleContent",
          children: children
        })]
      }), (actions || !showNavigationInTopArea && navigationActions) && /*#__PURE__*/_jsxs(Toolbar, {
        tabIndex: undefined,
        role: undefined,
        ...actionsToolbarProps,
        overflowButton: actionsToolbarProps?.overflowButton,
        design: "Auto",
        toolbarStyle: "Clear",
        active: true,
        className: clsx(classNames.toolbar, actionsToolbarProps?.className),
        onClick: handleActionsToolbarClick,
        "data-component-name": "ObjectPageTitleActions",
        onOverflowChange: actionsToolbarProps?.onOverflowChange,
        overflowPopoverRef: actionsOverflowRef,
        children: [/*#__PURE__*/_jsx(ActionsSpacer, {
          onClick: onHeaderClick,
          noHover: props?.['data-not-clickable']
        }), enhanceActionsWithClick(actions, syncedActionsOverflowRef), !showNavigationInTopArea && Children.count(actions) > 0 && Children.count(navigationActions) > 0 && /*#__PURE__*/_jsx(ToolbarSeparator, {}), !showNavigationInTopArea && enhanceActionsWithClick(navigationActions, syncedActionsOverflowRef)]
      })]
    }), subHeader && /*#__PURE__*/_jsx(FlexBox, {
      children: /*#__PURE__*/_jsx("div", {
        className: clsx(classNames.subTitle, classNames.subTitleBottom),
        "data-component-name": "ObjectPageTitleSubHeader",
        children: subHeader
      })
    }), props?.['data-header-content-visible'] ? expandedContent : props['data-is-snapped-rendered-outside'] ? undefined : snappedContent]
  });
});
ObjectPageTitle.displayName = 'ObjectPageTitle';
export { ObjectPageTitle };
