// @ui5/webcomponents-ngx/esm2022/main/table-header-row/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TableHeaderRow.js';
import * as i0 from "@angular/core";
let TableHeaderRowComponent = class TableHeaderRowComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableHeaderRowComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TableHeaderRowComponent, isStandalone: true, selector: "ui5-table-header-row", inputs: { sticky: ["sticky", "sticky", booleanAttribute] }, exportAs: ["ui5TableHeaderRow"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableHeaderRowComponent = __decorate([
    ProxyInputs(['sticky']),
    __metadata("design:paramtypes", [])
], TableHeaderRowComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableHeaderRowComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table-header-row',
                    template: '<ng-content></ng-content>',
                    inputs: ['sticky'],
                    exportAs: 'ui5TableHeaderRow',
                }]
        }], ctorParameters: () => [], propDecorators: { sticky: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { TableHeaderRowComponent };
