// @ui5/theming-ngx/esm2022/ui5-theming.service.mjs
import { Inject, Injectable, isDevMode, Optional } from '@angular/core';
import { BehaviorSubject, combineLatest, map, Observable, Subject, switchMap, takeUntil, tap, of, delay, } from 'rxjs';
import { UI5_THEMING_CONFIGURATION, } from './ui5-theming.models';
import * as i0 from "@angular/core";
export class Ui5ThemingService {
    constructor(_config) {
        this._config = _config;
        this._providers$ = new BehaviorSubject([]);
        this._currentTheme$ = new BehaviorSubject([
            this._config?.defaultTheme || 'sap_fiori_3',
            this._config?.defaultTheme || 'sap_fiori_3',
        ]);
        this._themeChanged$ = new Subject();
        this._destroyed$ = new Subject();
        combineLatest([
            this._providers$,
            this._currentTheme$.pipe(delay(100)) // Delay so that the providers are registered. @todo: Find a better solution
        ])
            .pipe(switchMap(([providers, [previousTheme, newTheme]]) => {
            return combineLatest(providers.map((provider) => {
                const isSupported = provider.supportsTheme(newTheme);
                if (typeof isSupported === 'boolean') {
                    return of([provider, isSupported]);
                }
                return isSupported.pipe(map((supported) => [provider, supported]));
            })).pipe(switchMap((providers) => {
                const unsupportedProviders = providers.filter(([, supported]) => !supported);
                if (unsupportedProviders.length) {
                    if (isDevMode()) {
                        console.warn(`The following providers do not support the theme "${newTheme}":`, unsupportedProviders.map(([provider]) => provider.name));
                    }
                }
                return combineLatest(providers.map(([provider]) => provider.setTheme(newTheme)));
            }), map((providerResponses) => {
                if (providerResponses.every(Boolean)) {
                    return newTheme;
                }
                return previousTheme;
            }));
        }), tap((theme) => {
            this._themeChanged$.next(theme);
        }), takeUntil(this._destroyed$))
            .subscribe();
    }
    supportsTheme(themeName) {
        return this.getAvailableThemes().pipe(map((themes) => themes.includes(themeName)));
    }
    getAvailableThemes() {
        return this._providers$.pipe(switchMap((providers) => {
            return combineLatest(providers.map((provider) => {
                const availableThemes = provider.getAvailableThemes();
                if (Array.isArray(availableThemes)) {
                    return of(availableThemes);
                }
                return availableThemes;
            })).pipe(map((providerResponses) => {
                const uniques = providerResponses.reduce((acc, curr) => {
                    curr.forEach((theme) => acc.add(theme));
                    return acc;
                }, new Set());
                return [...uniques.values()];
            }));
        }));
    }
    /** Registers theming provider for further usage. */
    registerProvider(provider) {
        const providers = this._providers$.value;
        providers.push(provider);
        this._providers$.next(providers);
    }
    /** Unregisters previously registered theming provider. */
    unregisterProvider(provider) {
        const providers = this._providers$.value;
        const index = providers.indexOf(provider);
        if (index !== -1) {
            providers.splice(index, 1);
            this._providers$.next(providers);
        }
    }
    /** Sets the theme to the providers */
    setTheme(theme) {
        return new Observable((subscriber) => {
            const finalizer$ = new Subject();
            this._themeChanged$
                .pipe(takeUntil(finalizer$))
                .subscribe((changedTheme) => {
                if (changedTheme === theme) {
                    finalizer$.next();
                    finalizer$.complete();
                }
                subscriber.next(changedTheme === theme);
                subscriber.complete();
            });
            this._currentTheme$.next([this._currentTheme$.value[1], theme]);
        });
    }
    /** @hidden */
    ngOnDestroy() {
        this._destroyed$.next();
        this._destroyed$.complete();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingService, deps: [{ token: UI5_THEMING_CONFIGURATION, optional: true }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [UI5_THEMING_CONFIGURATION]
                }] }] });
