// @ui5/theming-ngx/esm2022/ui5-theming.module.mjs
import { NgModule, Optional, isDevMode, } from '@angular/core';
import { Ui5ThemingService } from './ui5-theming.service';
import { UI5_THEMING_CONFIGURATION } from './ui5-theming.models';
import * as i0 from "@angular/core";
import * as i1 from "./ui5-theming.service";
export class Ui5ThemingModule {
    constructor(theming) {
        this.theming = theming;
        if (!this.theming && isDevMode()) {
            console.error('Could not find Ui5ThemingService. Make sure you called .forRoot() for Ui5ThemingModule');
        }
    }
    static forRoot(config) {
        return {
            ngModule: Ui5ThemingModule,
            providers: [
                Ui5ThemingService,
                {
                    provide: UI5_THEMING_CONFIGURATION,
                    useValue: config,
                },
            ],
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingModule, deps: [{ token: i1.Ui5ThemingService, optional: true }], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5ThemingModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [],
                    declarations: [],
                }]
        }], ctorParameters: () => [{ type: i1.Ui5ThemingService, decorators: [{
                    type: Optional
                }] }] });
