// @ui5/webcomponents-ngx/esm2022/main/card-header/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CardHeader.js';
import * as i0 from "@angular/core";
let CardHeaderComponent = class CardHeaderComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: CardHeaderComponent, isStandalone: true, selector: "ui5-card-header", inputs: { titleText: "titleText", subtitleText: "subtitleText", additionalText: "additionalText", interactive: ["interactive", "interactive", booleanAttribute] }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5CardHeader"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CardHeaderComponent = __decorate([
    ProxyInputs(['titleText', 'subtitleText', 'additionalText', 'interactive']),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], CardHeaderComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-card-header',
                    template: '<ng-content></ng-content>',
                    inputs: ['titleText', 'subtitleText', 'additionalText', 'interactive'],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5CardHeader',
                }]
        }], ctorParameters: () => [], propDecorators: { interactive: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { CardHeaderComponent };
