// @ui5/webcomponents-ngx/esm2022/main/label/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Label.js';
import * as i0 from "@angular/core";
let LabelComponent = class LabelComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: LabelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: LabelComponent, isStandalone: true, selector: "ui5-label", inputs: { for: "for", showColon: ["showColon", "showColon", booleanAttribute], required: ["required", "required", booleanAttribute], wrappingType: "wrappingType" }, exportAs: ["ui5Label"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
LabelComponent = __decorate([
    ProxyInputs(['for', 'showColon', 'required', 'wrappingType']),
    __metadata("design:paramtypes", [])
], LabelComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: LabelComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-label',
                    template: '<ng-content></ng-content>',
                    inputs: ['for', 'showColon', 'required', 'wrappingType'],
                    exportAs: 'ui5Label',
                }]
        }], ctorParameters: () => [], propDecorators: { showColon: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { LabelComponent };
