// @ui5/webcomponents-ngx/esm2022/main/calendar-legend/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CalendarLegend.js';
import * as i0 from "@angular/core";
let CalendarLegendComponent = class CalendarLegendComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarLegendComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: CalendarLegendComponent, isStandalone: true, selector: "ui5-calendar-legend", inputs: { hideToday: ["hideToday", "hideToday", booleanAttribute], hideSelectedDay: ["hideSelectedDay", "hideSelectedDay", booleanAttribute], hideNonWorkingDay: ["hideNonWorkingDay", "hideNonWorkingDay", booleanAttribute], hideWorkingDay: ["hideWorkingDay", "hideWorkingDay", booleanAttribute] }, exportAs: ["ui5CalendarLegend"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CalendarLegendComponent = __decorate([
    ProxyInputs([
        'hideToday',
        'hideSelectedDay',
        'hideNonWorkingDay',
        'hideWorkingDay',
    ]),
    __metadata("design:paramtypes", [])
], CalendarLegendComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarLegendComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-calendar-legend',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'hideToday',
                        'hideSelectedDay',
                        'hideNonWorkingDay',
                        'hideWorkingDay',
                    ],
                    exportAs: 'ui5CalendarLegend',
                }]
        }], ctorParameters: () => [], propDecorators: { hideToday: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideSelectedDay: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideNonWorkingDay: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideWorkingDay: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { CalendarLegendComponent };
