// @ui5/webcomponents-ngx/schematics/get-config.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.collectConfig = void 0;
const config_1 = require("./config");
const promt_1 = require("./utils/promt");
function collectConfig() {
    return __awaiter(this, void 0, void 0, function* () {
        const schema = {
            commonCss: [],
            theming: true,
            defaultTheme: 'sap_horizon'
        };
        const includeCommonCss = yield askCommonCssDependency();
        if (includeCommonCss) {
            schema.commonCss = yield askCommonCssParts();
        }
        return schema;
    });
}
exports.collectConfig = collectConfig;
function askCommonCssDependency() {
    return __awaiter(this, void 0, void 0, function* () {
        return yield (0, promt_1.askQuestion)({
            type: 'confirm',
            message: 'Would you like to add Common CSS into your application?',
            default: false,
        });
    });
}
function askCommonCssParts() {
    return __awaiter(this, void 0, void 0, function* () {
        return yield (0, promt_1.askQuestion)({
            type: 'checkbox',
            message: 'Please select desired Common CSS features',
            choices: config_1.CommonCssParts,
        });
    });
}
