// @ui5/webcomponents-ngx/esm2022/main/table-header-cell/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TableHeaderCell.js';
import * as i0 from "@angular/core";
let TableHeaderCellComponent = class TableHeaderCellComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableHeaderCellComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TableHeaderCellComponent, isStandalone: true, selector: "ui5-table-header-cell", inputs: { width: "width", minWidth: "minWidth", maxWidth: "maxWidth", importance: "importance" }, exportAs: ["ui5TableHeaderCell"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableHeaderCellComponent = __decorate([
    ProxyInputs(['width', 'minWidth', 'maxWidth', 'importance']),
    __metadata("design:paramtypes", [])
], TableHeaderCellComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableHeaderCellComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table-header-cell',
                    template: '<ng-content></ng-content>',
                    inputs: ['width', 'minWidth', 'maxWidth', 'importance'],
                    exportAs: 'ui5TableHeaderCell',
                }]
        }], ctorParameters: () => [] });
export { TableHeaderCellComponent };
