// @ui5/webcomponents-ngx/esm2022/main/toolbar-button/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarButton.js';
import * as i0 from "@angular/core";
let ToolbarButtonComponent = class ToolbarButtonComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarButtonComponent, isStandalone: true, selector: "ui5-toolbar-button", inputs: { overflowPriority: "overflowPriority", preventOverflowClosing: ["preventOverflowClosing", "preventOverflowClosing", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], design: "design", icon: "icon", endIcon: "endIcon", tooltip: "tooltip", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibilityAttributes: "accessibilityAttributes", text: "text", width: "width" }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5ToolbarButton"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarButtonComponent = __decorate([
    ProxyInputs([
        'overflowPriority',
        'preventOverflowClosing',
        'disabled',
        'design',
        'icon',
        'endIcon',
        'tooltip',
        'accessibleName',
        'accessibleNameRef',
        'accessibilityAttributes',
        'text',
        'width',
    ]),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], ToolbarButtonComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarButtonComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-button',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'overflowPriority',
                        'preventOverflowClosing',
                        'disabled',
                        'design',
                        'icon',
                        'endIcon',
                        'tooltip',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibilityAttributes',
                        'text',
                        'width',
                    ],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5ToolbarButton',
                }]
        }], ctorParameters: () => [], propDecorators: { preventOverflowClosing: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ToolbarButtonComponent };
