// @ui5/webcomponents-ngx/esm2022/main/toast/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Toast.js';
import * as i0 from "@angular/core";
let ToastComponent = class ToastComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToastComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToastComponent, isStandalone: true, selector: "ui5-toast", inputs: { duration: "duration", placement: "placement", open: ["open", "open", booleanAttribute] }, outputs: { ui5Close: "ui5Close" }, exportAs: ["ui5Toast"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToastComponent = __decorate([
    ProxyInputs(['duration', 'placement', 'open']),
    ProxyOutputs(['close: ui5Close']),
    __metadata("design:paramtypes", [])
], ToastComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToastComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toast',
                    template: '<ng-content></ng-content>',
                    inputs: ['duration', 'placement', 'open'],
                    outputs: ['ui5Close'],
                    exportAs: 'ui5Toast',
                }]
        }], ctorParameters: () => [], propDecorators: { open: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ToastComponent };
