// @ui5/webcomponents-ngx/esm2022/fiori/notification-list/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/NotificationList.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let NotificationListComponent = class NotificationListComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: NotificationListComponent, isStandalone: true, selector: "ui5-notification-list", inputs: { noDataText: "noDataText" }, outputs: { ui5ItemClick: "ui5ItemClick", ui5ItemClose: "ui5ItemClose", ui5ItemToggle: "ui5ItemToggle" }, exportAs: ["ui5NotificationList"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
NotificationListComponent = __decorate([
    ProxyInputs(['noDataText']),
    ProxyOutputs([
        'item-click: ui5ItemClick',
        'item-close: ui5ItemClose',
        'item-toggle: ui5ItemToggle',
    ]),
    __metadata("design:paramtypes", [])
], NotificationListComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-notification-list',
                    template: '<ng-content></ng-content>',
                    inputs: ['noDataText'],
                    outputs: ['ui5ItemClick', 'ui5ItemClose', 'ui5ItemToggle'],
                    exportAs: 'ui5NotificationList',
                }]
        }], ctorParameters: () => [] });
export { NotificationListComponent };
