// @ui5/webcomponents-ngx/esm2022/main/form-group/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/FormGroup.js';
import * as i0 from "@angular/core";
let FormGroupComponent = class FormGroupComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: FormGroupComponent, isStandalone: true, selector: "ui5-form-group", inputs: { headerText: "headerText", columnSpan: "columnSpan" }, exportAs: ["ui5FormGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FormGroupComponent = __decorate([
    ProxyInputs(['headerText', 'columnSpan']),
    __metadata("design:paramtypes", [])
], FormGroupComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormGroupComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-form-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['headerText', 'columnSpan'],
                    exportAs: 'ui5FormGroup',
                }]
        }], ctorParameters: () => [] });
export { FormGroupComponent };
