// @ui5/webcomponents-ngx/esm2022/content-density/index.mjs
import { Directive, HostBinding, Input, NgModule } from "@angular/core";
import * as i0 from "@angular/core";
export class CompactDirective {
    constructor() {
        this._ui5Compact = false;
    }
    set ui5Compact(value) {
        this._ui5Compact = value === true || value === '' || value === 'true';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CompactDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.0.9", type: CompactDirective, isStandalone: true, selector: "[ui5Compact]", inputs: { ui5Compact: "ui5Compact" }, host: { properties: { "class.ui5-content-density-compact": "this._ui5Compact" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CompactDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[ui5Compact]',
                    standalone: true
                }]
        }], propDecorators: { ui5Compact: [{
                type: Input
            }], _ui5Compact: [{
                type: HostBinding,
                args: ['class.ui5-content-density-compact']
            }] } });
export class ContentDensityModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ContentDensityModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: ContentDensityModule, imports: [CompactDirective], exports: [CompactDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ContentDensityModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ContentDensityModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CompactDirective],
                    exports: [CompactDirective]
                }]
        }] });
