// @ui5/webcomponents-ngx/esm2022/fiori/sort-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/SortItem.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let SortItemComponent = class SortItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SortItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SortItemComponent, isStandalone: true, selector: "ui5-sort-item", inputs: { text: "text", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5SortItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SortItemComponent = __decorate([
    ProxyInputs(['text', 'selected']),
    __metadata("design:paramtypes", [])
], SortItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SortItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-sort-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'selected'],
                    exportAs: 'ui5SortItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { SortItemComponent };
