// @ui5/webcomponents-ngx/esm2022/main/responsive-popover/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ResponsivePopover.js';
import * as i0 from "@angular/core";
let ResponsivePopoverComponent = class ResponsivePopoverComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ResponsivePopoverComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ResponsivePopoverComponent, isStandalone: true, selector: "ui5-responsive-popover", inputs: { initialFocus: "initialFocus", preventFocusRestore: ["preventFocusRestore", "preventFocusRestore", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibleRole: "accessibleRole", preventInitialFocus: ["preventInitialFocus", "preventInitialFocus", booleanAttribute], open: ["open", "open", booleanAttribute], headerText: "headerText", placement: "placement", horizontalAlign: "horizontalAlign", verticalAlign: "verticalAlign", modal: ["modal", "modal", booleanAttribute], hideArrow: ["hideArrow", "hideArrow", booleanAttribute], allowTargetOverlap: ["allowTargetOverlap", "allowTargetOverlap", booleanAttribute], opener: "opener" }, outputs: { ui5BeforeOpen: "ui5BeforeOpen", ui5Open: "ui5Open", ui5BeforeClose: "ui5BeforeClose", ui5Close: "ui5Close" }, exportAs: ["ui5ResponsivePopover"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ResponsivePopoverComponent = __decorate([
    ProxyInputs([
        'initialFocus',
        'preventFocusRestore',
        'accessibleName',
        'accessibleNameRef',
        'accessibleRole',
        'preventInitialFocus',
        'open',
        'headerText',
        'placement',
        'horizontalAlign',
        'verticalAlign',
        'modal',
        'hideArrow',
        'allowTargetOverlap',
        'opener',
    ]),
    ProxyOutputs([
        'before-open: ui5BeforeOpen',
        'open: ui5Open',
        'before-close: ui5BeforeClose',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], ResponsivePopoverComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ResponsivePopoverComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-responsive-popover',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'initialFocus',
                        'preventFocusRestore',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibleRole',
                        'preventInitialFocus',
                        'open',
                        'headerText',
                        'placement',
                        'horizontalAlign',
                        'verticalAlign',
                        'modal',
                        'hideArrow',
                        'allowTargetOverlap',
                        'opener',
                    ],
                    outputs: ['ui5BeforeOpen', 'ui5Open', 'ui5BeforeClose', 'ui5Close'],
                    exportAs: 'ui5ResponsivePopover',
                }]
        }], ctorParameters: () => [], propDecorators: { preventFocusRestore: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], preventInitialFocus: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], modal: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideArrow: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], allowTargetOverlap: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ResponsivePopoverComponent };
