// @ui5/webcomponents-ngx/esm2022/fiori/timeline/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/Timeline.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let TimelineComponent = class TimelineComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TimelineComponent, isStandalone: true, selector: "ui5-timeline", inputs: { layout: "layout", accessibleName: "accessibleName" }, exportAs: ["ui5Timeline"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TimelineComponent = __decorate([
    ProxyInputs(['layout', 'accessibleName']),
    __metadata("design:paramtypes", [])
], TimelineComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-timeline',
                    template: '<ng-content></ng-content>',
                    inputs: ['layout', 'accessibleName'],
                    exportAs: 'ui5Timeline',
                }]
        }], ctorParameters: () => [] });
export { TimelineComponent };
