// @ui5/webcomponents-ngx/esm2022/main/option-custom/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/OptionCustom.js';
import * as i0 from "@angular/core";
let OptionCustomComponent = class OptionCustomComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: OptionCustomComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: OptionCustomComponent, isStandalone: true, selector: "ui5-option-custom", inputs: { displayText: "displayText", value: "value", tooltip: "tooltip", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5OptionCustom"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
OptionCustomComponent = __decorate([
    ProxyInputs(['displayText', 'value', 'tooltip', 'selected']),
    __metadata("design:paramtypes", [])
], OptionCustomComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: OptionCustomComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-option-custom',
                    template: '<ng-content></ng-content>',
                    inputs: ['displayText', 'value', 'tooltip', 'selected'],
                    exportAs: 'ui5OptionCustom',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { OptionCustomComponent };
