// @ui5/webcomponents-ngx/esm2022/fiori/upload-collection/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/UploadCollection.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let UploadCollectionComponent = class UploadCollectionComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: UploadCollectionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: UploadCollectionComponent, isStandalone: true, selector: "ui5-upload-collection", inputs: { selectionMode: "selectionMode", noDataDescription: "noDataDescription", noDataText: "noDataText", hideDragOverlay: ["hideDragOverlay", "hideDragOverlay", booleanAttribute], accessibleName: "accessibleName" }, outputs: { ui5Drop: "ui5Drop", ui5ItemDelete: "ui5ItemDelete", ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5UploadCollection"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
UploadCollectionComponent = __decorate([
    ProxyInputs([
        'selectionMode',
        'noDataDescription',
        'noDataText',
        'hideDragOverlay',
        'accessibleName',
    ]),
    ProxyOutputs([
        'drop: ui5Drop',
        'item-delete: ui5ItemDelete',
        'selection-change: ui5SelectionChange',
    ]),
    __metadata("design:paramtypes", [])
], UploadCollectionComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: UploadCollectionComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-upload-collection',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'selectionMode',
                        'noDataDescription',
                        'noDataText',
                        'hideDragOverlay',
                        'accessibleName',
                    ],
                    outputs: ['ui5Drop', 'ui5ItemDelete', 'ui5SelectionChange'],
                    exportAs: 'ui5UploadCollection',
                }]
        }], ctorParameters: () => [], propDecorators: { hideDragOverlay: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { UploadCollectionComponent };
