// @ui5/webcomponents-ngx/esm2022/main/color-palette-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ColorPaletteItem.js';
import * as i0 from "@angular/core";
let ColorPaletteItemComponent = class ColorPaletteItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ColorPaletteItemComponent, isStandalone: true, selector: "ui5-color-palette-item", inputs: { value: "value", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5ColorPaletteItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ColorPaletteItemComponent = __decorate([
    ProxyInputs(['value', 'selected']),
    __metadata("design:paramtypes", [])
], ColorPaletteItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-color-palette-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['value', 'selected'],
                    exportAs: 'ui5ColorPaletteItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ColorPaletteItemComponent };
