// @ui5/webcomponents-ngx/esm2022/main/card/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Card.js';
import * as i0 from "@angular/core";
let CardComponent = class CardComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: CardComponent, isStandalone: true, selector: "ui5-card", inputs: { accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, exportAs: ["ui5Card"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CardComponent = __decorate([
    ProxyInputs(['accessibleName', 'accessibleNameRef']),
    __metadata("design:paramtypes", [])
], CardComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-card',
                    template: '<ng-content></ng-content>',
                    inputs: ['accessibleName', 'accessibleNameRef'],
                    exportAs: 'ui5Card',
                }]
        }], ctorParameters: () => [] });
export { CardComponent };
