// @ui5/webcomponents-ngx/esm2022/main/link/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Link.js';
import * as i0 from "@angular/core";
let LinkComponent = class LinkComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: LinkComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: LinkComponent, isStandalone: true, selector: "ui5-link", inputs: { disabled: ["disabled", "disabled", booleanAttribute], tooltip: "tooltip", href: "href", target: "target", design: "design", wrappingType: "wrappingType", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibleRole: "accessibleRole", accessibilityAttributes: "accessibilityAttributes", icon: "icon", endIcon: "endIcon" }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5Link"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
LinkComponent = __decorate([
    ProxyInputs([
        'disabled',
        'tooltip',
        'href',
        'target',
        'design',
        'wrappingType',
        'accessibleName',
        'accessibleNameRef',
        'accessibleRole',
        'accessibilityAttributes',
        'icon',
        'endIcon',
    ]),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], LinkComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: LinkComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-link',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'tooltip',
                        'href',
                        'target',
                        'design',
                        'wrappingType',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibleRole',
                        'accessibilityAttributes',
                        'icon',
                        'endIcon',
                    ],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5Link',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { LinkComponent };
