// @ui5/webcomponents-ngx/esm2022/main/menu-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/MenuItem.js';
import * as i0 from "@angular/core";
let MenuItemComponent = class MenuItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MenuItemComponent, isStandalone: true, selector: "ui5-menu-item", inputs: { type: "type", accessibilityAttributes: "accessibilityAttributes", navigated: ["navigated", "navigated", booleanAttribute], tooltip: "tooltip", highlight: "highlight", selected: ["selected", "selected", booleanAttribute], text: "text", additionalText: "additionalText", icon: "icon", disabled: ["disabled", "disabled", booleanAttribute], loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay", accessibleName: "accessibleName" }, outputs: { ui5DetailClick: "ui5DetailClick" }, exportAs: ["ui5MenuItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MenuItemComponent = __decorate([
    ProxyInputs([
        'type',
        'accessibilityAttributes',
        'navigated',
        'tooltip',
        'highlight',
        'selected',
        'text',
        'additionalText',
        'icon',
        'disabled',
        'loading',
        'loadingDelay',
        'accessibleName',
    ]),
    ProxyOutputs(['detail-click: ui5DetailClick']),
    __metadata("design:paramtypes", [])
], MenuItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-menu-item',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'type',
                        'accessibilityAttributes',
                        'navigated',
                        'tooltip',
                        'highlight',
                        'selected',
                        'text',
                        'additionalText',
                        'icon',
                        'disabled',
                        'loading',
                        'loadingDelay',
                        'accessibleName',
                    ],
                    outputs: ['ui5DetailClick'],
                    exportAs: 'ui5MenuItem',
                }]
        }], ctorParameters: () => [], propDecorators: { navigated: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], loading: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { MenuItemComponent };
