// @ui5/webcomponents-ngx/esm2022/main/tag/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Tag.js';
import * as i0 from "@angular/core";
let TagComponent = class TagComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TagComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TagComponent, isStandalone: true, selector: "ui5-tag", inputs: { design: "design", colorScheme: "colorScheme", hideStateIcon: ["hideStateIcon", "hideStateIcon", booleanAttribute], interactive: ["interactive", "interactive", booleanAttribute], wrappingType: "wrappingType", size: "size" }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5Tag"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TagComponent = __decorate([
    ProxyInputs([
        'design',
        'colorScheme',
        'hideStateIcon',
        'interactive',
        'wrappingType',
        'size',
    ]),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], TagComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TagComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tag',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'design',
                        'colorScheme',
                        'hideStateIcon',
                        'interactive',
                        'wrappingType',
                        'size',
                    ],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5Tag',
                }]
        }], ctorParameters: () => [], propDecorators: { hideStateIcon: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], interactive: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { TagComponent };
