// @ui5/webcomponents-react/dist/internal/ModalStore.js
const STORE_SYMBOL_LISTENERS = Symbol.for('@ui5/webcomponents-react/Modals/Listeners');
const STORE_SYMBOL = Symbol.for('@ui5/webcomponents-react/Modals');
const initialStore = [];
function getListeners() {
  globalThis[STORE_SYMBOL_LISTENERS] ??= [];
  return globalThis[STORE_SYMBOL_LISTENERS];
}
function emitChange() {
  for (const listener of getListeners()) {
    listener();
  }
}
function getSnapshot() {
  globalThis[STORE_SYMBOL] ??= initialStore;
  return globalThis[STORE_SYMBOL];
}
function subscribe(listener) {
  const listeners = getListeners();
  globalThis[STORE_SYMBOL_LISTENERS] = [...listeners, listener];
  return () => {
    globalThis[STORE_SYMBOL_LISTENERS] = listeners.filter(l => l !== listener);
  };
}
export const ModalStore = {
  subscribe,
  getSnapshot,
  getServerSnapshot: () => {
    return initialStore;
  },
  addModal(config) {
    globalThis[STORE_SYMBOL] = [...getSnapshot(), config];
    emitChange();
  },
  removeModal(id) {
    globalThis[STORE_SYMBOL] = getSnapshot().filter(modal => modal.id !== id);
    emitChange();
  }
};
