// @ui5/webcomponents-ngx/esm2022/fiori/side-navigation-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/SideNavigationItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let SideNavigationItemComponent = class SideNavigationItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SideNavigationItemComponent, isStandalone: true, selector: "ui5-side-navigation-item", inputs: { text: "text", disabled: ["disabled", "disabled", booleanAttribute], tooltip: "tooltip", icon: "icon", selected: ["selected", "selected", booleanAttribute], href: "href", target: "target", expanded: ["expanded", "expanded", booleanAttribute] }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5SideNavigationItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SideNavigationItemComponent = __decorate([
    ProxyInputs([
        'text',
        'disabled',
        'tooltip',
        'icon',
        'selected',
        'href',
        'target',
        'expanded',
    ]),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], SideNavigationItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-side-navigation-item',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'text',
                        'disabled',
                        'tooltip',
                        'icon',
                        'selected',
                        'href',
                        'target',
                        'expanded',
                    ],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5SideNavigationItem',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], expanded: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { SideNavigationItemComponent };
