// @ui5/webcomponents-ngx/esm2022/fiori/wizard/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/Wizard.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let WizardComponent = class WizardComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WizardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: WizardComponent, isStandalone: true, selector: "ui5-wizard", inputs: { contentLayout: "contentLayout" }, outputs: { ui5StepChange: "ui5StepChange" }, exportAs: ["ui5Wizard"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
WizardComponent = __decorate([
    ProxyInputs(['contentLayout']),
    ProxyOutputs(['step-change: ui5StepChange']),
    __metadata("design:paramtypes", [])
], WizardComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WizardComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-wizard',
                    template: '<ng-content></ng-content>',
                    inputs: ['contentLayout'],
                    outputs: ['ui5StepChange'],
                    exportAs: 'ui5Wizard',
                }]
        }], ctorParameters: () => [] });
export { WizardComponent };
