// @ui5/webcomponents-ngx/esm2022/fiori/barcode-scanner-dialog/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/BarcodeScannerDialog.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let BarcodeScannerDialogComponent = class BarcodeScannerDialogComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BarcodeScannerDialogComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: BarcodeScannerDialogComponent, isStandalone: true, selector: "ui5-barcode-scanner-dialog", inputs: { open: ["open", "open", booleanAttribute] }, outputs: { ui5Close: "ui5Close", ui5ScanSuccess: "ui5ScanSuccess", ui5ScanError: "ui5ScanError" }, exportAs: ["ui5BarcodeScannerDialog"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
BarcodeScannerDialogComponent = __decorate([
    ProxyInputs(['open']),
    ProxyOutputs([
        'close: ui5Close',
        'scan-success: ui5ScanSuccess',
        'scan-error: ui5ScanError',
    ]),
    __metadata("design:paramtypes", [])
], BarcodeScannerDialogComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BarcodeScannerDialogComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-barcode-scanner-dialog',
                    template: '<ng-content></ng-content>',
                    inputs: ['open'],
                    outputs: ['ui5Close', 'ui5ScanSuccess', 'ui5ScanError'],
                    exportAs: 'ui5BarcodeScannerDialog',
                }]
        }], ctorParameters: () => [], propDecorators: { open: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { BarcodeScannerDialogComponent };
