// @ui5/webcomponents-ngx/esm2022/fiori/upload-collection-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/UploadCollectionItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let UploadCollectionItemComponent = class UploadCollectionItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: UploadCollectionItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: UploadCollectionItemComponent, isStandalone: true, selector: "ui5-upload-collection-item", inputs: { type: "type", accessibilityAttributes: "accessibilityAttributes", navigated: ["navigated", "navigated", booleanAttribute], tooltip: "tooltip", highlight: "highlight", selected: ["selected", "selected", booleanAttribute], file: "file", fileName: "fileName", fileNameClickable: ["fileNameClickable", "fileNameClickable", booleanAttribute], disableDeleteButton: ["disableDeleteButton", "disableDeleteButton", booleanAttribute], hideDeleteButton: ["hideDeleteButton", "hideDeleteButton", booleanAttribute], hideRetryButton: ["hideRetryButton", "hideRetryButton", booleanAttribute], hideTerminateButton: ["hideTerminateButton", "hideTerminateButton", booleanAttribute], progress: "progress", uploadState: "uploadState" }, outputs: { ui5DetailClick: "ui5DetailClick", ui5FileNameClick: "ui5FileNameClick", ui5Rename: "ui5Rename", ui5Terminate: "ui5Terminate", ui5Retry: "ui5Retry" }, exportAs: ["ui5UploadCollectionItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
UploadCollectionItemComponent = __decorate([
    ProxyInputs([
        'type',
        'accessibilityAttributes',
        'navigated',
        'tooltip',
        'highlight',
        'selected',
        'file',
        'fileName',
        'fileNameClickable',
        'disableDeleteButton',
        'hideDeleteButton',
        'hideRetryButton',
        'hideTerminateButton',
        'progress',
        'uploadState',
    ]),
    ProxyOutputs([
        'detail-click: ui5DetailClick',
        'file-name-click: ui5FileNameClick',
        'rename: ui5Rename',
        'terminate: ui5Terminate',
        'retry: ui5Retry',
    ]),
    __metadata("design:paramtypes", [])
], UploadCollectionItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: UploadCollectionItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-upload-collection-item',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'type',
                        'accessibilityAttributes',
                        'navigated',
                        'tooltip',
                        'highlight',
                        'selected',
                        'file',
                        'fileName',
                        'fileNameClickable',
                        'disableDeleteButton',
                        'hideDeleteButton',
                        'hideRetryButton',
                        'hideTerminateButton',
                        'progress',
                        'uploadState',
                    ],
                    outputs: [
                        'ui5DetailClick',
                        'ui5FileNameClick',
                        'ui5Rename',
                        'ui5Terminate',
                        'ui5Retry',
                    ],
                    exportAs: 'ui5UploadCollectionItem',
                }]
        }], ctorParameters: () => [], propDecorators: { navigated: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], fileNameClickable: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disableDeleteButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideDeleteButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideRetryButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideTerminateButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { UploadCollectionItemComponent };
