// @ui5/webcomponents-ngx/esm2022/fiori/page/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/Page.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let PageComponent = class PageComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: PageComponent, isStandalone: true, selector: "ui5-page", inputs: { backgroundDesign: "backgroundDesign", noScrolling: ["noScrolling", "noScrolling", booleanAttribute], fixedFooter: ["fixedFooter", "fixedFooter", booleanAttribute], hideFooter: ["hideFooter", "hideFooter", booleanAttribute] }, exportAs: ["ui5Page"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
PageComponent = __decorate([
    ProxyInputs(['backgroundDesign', 'noScrolling', 'fixedFooter', 'hideFooter']),
    __metadata("design:paramtypes", [])
], PageComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PageComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-page',
                    template: '<ng-content></ng-content>',
                    inputs: ['backgroundDesign', 'noScrolling', 'fixedFooter', 'hideFooter'],
                    exportAs: 'ui5Page',
                }]
        }], ctorParameters: () => [], propDecorators: { noScrolling: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], fixedFooter: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideFooter: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { PageComponent };
