// @ui5/webcomponents-ngx/esm2022/main/bar/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Bar.js';
import * as i0 from "@angular/core";
let BarComponent = class BarComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: BarComponent, isStandalone: true, selector: "ui5-bar", inputs: { design: "design" }, exportAs: ["ui5Bar"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
BarComponent = __decorate([
    ProxyInputs(['design']),
    __metadata("design:paramtypes", [])
], BarComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BarComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-bar',
                    template: '<ng-content></ng-content>',
                    inputs: ['design'],
                    exportAs: 'ui5Bar',
                }]
        }], ctorParameters: () => [] });
export { BarComponent };
