// @ui5/webcomponents-ngx/esm2022/main/toolbar-spacer/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarSpacer.js';
import * as i0 from "@angular/core";
let ToolbarSpacerComponent = class ToolbarSpacerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSpacerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarSpacerComponent, isStandalone: true, selector: "ui5-toolbar-spacer", inputs: { overflowPriority: "overflowPriority", preventOverflowClosing: ["preventOverflowClosing", "preventOverflowClosing", booleanAttribute], width: "width" }, exportAs: ["ui5ToolbarSpacer"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarSpacerComponent = __decorate([
    ProxyInputs(['overflowPriority', 'preventOverflowClosing', 'width']),
    __metadata("design:paramtypes", [])
], ToolbarSpacerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSpacerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-spacer',
                    template: '<ng-content></ng-content>',
                    inputs: ['overflowPriority', 'preventOverflowClosing', 'width'],
                    exportAs: 'ui5ToolbarSpacer',
                }]
        }], ctorParameters: () => [], propDecorators: { preventOverflowClosing: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ToolbarSpacerComponent };
