// @ui5/webcomponents-ngx/esm2022/main/list-item-custom/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ListItemCustom.js';
import * as i0 from "@angular/core";
let ListItemCustomComponent = class ListItemCustomComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListItemCustomComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ListItemCustomComponent, isStandalone: true, selector: "ui5-li-custom", inputs: { type: "type", accessibilityAttributes: "accessibilityAttributes", navigated: ["navigated", "navigated", booleanAttribute], tooltip: "tooltip", highlight: "highlight", selected: ["selected", "selected", booleanAttribute], movable: ["movable", "movable", booleanAttribute], accessibleName: "accessibleName" }, outputs: { ui5DetailClick: "ui5DetailClick" }, exportAs: ["ui5LiCustom"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ListItemCustomComponent = __decorate([
    ProxyInputs([
        'type',
        'accessibilityAttributes',
        'navigated',
        'tooltip',
        'highlight',
        'selected',
        'movable',
        'accessibleName',
    ]),
    ProxyOutputs(['detail-click: ui5DetailClick']),
    __metadata("design:paramtypes", [])
], ListItemCustomComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListItemCustomComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-li-custom',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'type',
                        'accessibilityAttributes',
                        'navigated',
                        'tooltip',
                        'highlight',
                        'selected',
                        'movable',
                        'accessibleName',
                    ],
                    outputs: ['ui5DetailClick'],
                    exportAs: 'ui5LiCustom',
                }]
        }], ctorParameters: () => [], propDecorators: { navigated: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], movable: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ListItemCustomComponent };
