// @ui5/webcomponents-ngx/esm2022/main/list-item-group/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ListItemGroup.js';
import * as i0 from "@angular/core";
let ListItemGroupComponent = class ListItemGroupComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListItemGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ListItemGroupComponent, isStandalone: true, selector: "ui5-li-group", inputs: { headerText: "headerText", headerAccessibleName: "headerAccessibleName" }, exportAs: ["ui5LiGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ListItemGroupComponent = __decorate([
    ProxyInputs(['headerText', 'headerAccessibleName']),
    __metadata("design:paramtypes", [])
], ListItemGroupComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListItemGroupComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-li-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['headerText', 'headerAccessibleName'],
                    exportAs: 'ui5LiGroup',
                }]
        }], ctorParameters: () => [] });
export { ListItemGroupComponent };
