// @ui5/webcomponents-ngx/esm2022/fiori/filter-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/FilterItem.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let FilterItemComponent = class FilterItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FilterItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: FilterItemComponent, isStandalone: true, selector: "ui5-filter-item", inputs: { text: "text", additionalText: "additionalText" }, exportAs: ["ui5FilterItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FilterItemComponent = __decorate([
    ProxyInputs(['text', 'additionalText']),
    __metadata("design:paramtypes", [])
], FilterItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FilterItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-filter-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'additionalText'],
                    exportAs: 'ui5FilterItem',
                }]
        }], ctorParameters: () => [] });
export { FilterItemComponent };
