// @ui5/webcomponents-ngx/esm2022/fiori/dynamic-side-content/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/DynamicSideContent.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let DynamicSideContentComponent = class DynamicSideContentComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicSideContentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: DynamicSideContentComponent, isStandalone: true, selector: "ui5-dynamic-side-content", inputs: { hideMainContent: ["hideMainContent", "hideMainContent", booleanAttribute], hideSideContent: ["hideSideContent", "hideSideContent", booleanAttribute], sideContentPosition: "sideContentPosition", sideContentVisibility: "sideContentVisibility", sideContentFallDown: "sideContentFallDown", equalSplit: ["equalSplit", "equalSplit", booleanAttribute] }, outputs: { ui5LayoutChange: "ui5LayoutChange" }, exportAs: ["ui5DynamicSideContent"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
DynamicSideContentComponent = __decorate([
    ProxyInputs([
        'hideMainContent',
        'hideSideContent',
        'sideContentPosition',
        'sideContentVisibility',
        'sideContentFallDown',
        'equalSplit',
    ]),
    ProxyOutputs(['layout-change: ui5LayoutChange']),
    __metadata("design:paramtypes", [])
], DynamicSideContentComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicSideContentComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-dynamic-side-content',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'hideMainContent',
                        'hideSideContent',
                        'sideContentPosition',
                        'sideContentVisibility',
                        'sideContentFallDown',
                        'equalSplit',
                    ],
                    outputs: ['ui5LayoutChange'],
                    exportAs: 'ui5DynamicSideContent',
                }]
        }], ctorParameters: () => [], propDecorators: { hideMainContent: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], hideSideContent: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], equalSplit: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { DynamicSideContentComponent };
