// @ui5/webcomponents-ngx/esm2022/main/color-palette-popover/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ColorPalettePopover.js';
import * as i0 from "@angular/core";
let ColorPalettePopoverComponent = class ColorPalettePopoverComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPalettePopoverComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ColorPalettePopoverComponent, isStandalone: true, selector: "ui5-color-palette-popover", inputs: { showRecentColors: ["showRecentColors", "showRecentColors", booleanAttribute], showMoreColors: ["showMoreColors", "showMoreColors", booleanAttribute], showDefaultColor: ["showDefaultColor", "showDefaultColor", booleanAttribute], defaultColor: "defaultColor", open: ["open", "open", booleanAttribute], opener: "opener" }, outputs: { ui5ItemClick: "ui5ItemClick", ui5Close: "ui5Close" }, exportAs: ["ui5ColorPalettePopover"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ColorPalettePopoverComponent = __decorate([
    ProxyInputs([
        'showRecentColors',
        'showMoreColors',
        'showDefaultColor',
        'defaultColor',
        'open',
        'opener',
    ]),
    ProxyOutputs(['item-click: ui5ItemClick', 'close: ui5Close']),
    __metadata("design:paramtypes", [])
], ColorPalettePopoverComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPalettePopoverComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-color-palette-popover',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'showRecentColors',
                        'showMoreColors',
                        'showDefaultColor',
                        'defaultColor',
                        'open',
                        'opener',
                    ],
                    outputs: ['ui5ItemClick', 'ui5Close'],
                    exportAs: 'ui5ColorPalettePopover',
                }]
        }], ctorParameters: () => [], propDecorators: { showRecentColors: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showMoreColors: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showDefaultColor: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ColorPalettePopoverComponent };
