// @ui5/webcomponents-ngx/esm2022/main/special-calendar-date/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SpecialCalendarDate.js';
import * as i0 from "@angular/core";
let SpecialCalendarDateComponent = class SpecialCalendarDateComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SpecialCalendarDateComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: SpecialCalendarDateComponent, isStandalone: true, selector: "ui5-special-date", inputs: { value: "value", type: "type" }, exportAs: ["ui5SpecialDate"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SpecialCalendarDateComponent = __decorate([
    ProxyInputs(['value', 'type']),
    __metadata("design:paramtypes", [])
], SpecialCalendarDateComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SpecialCalendarDateComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-special-date',
                    template: '<ng-content></ng-content>',
                    inputs: ['value', 'type'],
                    exportAs: 'ui5SpecialDate',
                }]
        }], ctorParameters: () => [] });
export { SpecialCalendarDateComponent };
