// @ui5/webcomponents-ngx/esm2022/fiori/side-navigation/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/SideNavigation.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let SideNavigationComponent = class SideNavigationComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SideNavigationComponent, isStandalone: true, selector: "ui5-side-navigation", inputs: { collapsed: ["collapsed", "collapsed", booleanAttribute] }, outputs: { ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5SideNavigation"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SideNavigationComponent = __decorate([
    ProxyInputs(['collapsed']),
    ProxyOutputs(['selection-change: ui5SelectionChange']),
    __metadata("design:paramtypes", [])
], SideNavigationComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-side-navigation',
                    template: '<ng-content></ng-content>',
                    inputs: ['collapsed'],
                    outputs: ['ui5SelectionChange'],
                    exportAs: 'ui5SideNavigation',
                }]
        }], ctorParameters: () => [], propDecorators: { collapsed: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { SideNavigationComponent };
