// @ui5/webcomponents-ngx/esm2022/main/slider/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Slider.js';
import * as i0 from "@angular/core";
import * as i1 from "@ui5/webcomponents-ngx/generic-cva";
let SliderComponent = class SliderComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SliderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SliderComponent, isStandalone: true, selector: "ui5-slider", inputs: { min: "min", max: "max", name: "name", step: "step", labelInterval: "labelInterval", showTickmarks: ["showTickmarks", "showTickmarks", booleanAttribute], showTooltip: ["showTooltip", "showTooltip", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], accessibleName: "accessibleName", value: "value" }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5Slider"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SliderComponent = __decorate([
    ProxyInputs([
        'min',
        'max',
        'name',
        'step',
        'labelInterval',
        'showTickmarks',
        'showTooltip',
        'disabled',
        'accessibleName',
        'value',
    ]),
    ProxyOutputs(['change: ui5Change', 'input: ui5Input']),
    __metadata("design:paramtypes", [])
], SliderComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SliderComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-slider',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'min',
                        'max',
                        'name',
                        'step',
                        'labelInterval',
                        'showTickmarks',
                        'showTooltip',
                        'disabled',
                        'accessibleName',
                        'value',
                    ],
                    outputs: ['ui5Change', 'ui5Input'],
                    exportAs: 'ui5Slider',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { showTickmarks: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showTooltip: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { SliderComponent };
