// @ui5/webcomponents-ngx/esm2022/fiori/dynamic-page/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/DynamicPage.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let DynamicPageComponent = class DynamicPageComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicPageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: DynamicPageComponent, isStandalone: true, selector: "ui5-dynamic-page", inputs: { hidePinButton: ["hidePinButton", "hidePinButton", booleanAttribute], headerPinned: ["headerPinned", "headerPinned", booleanAttribute], showFooter: ["showFooter", "showFooter", booleanAttribute], headerSnapped: ["headerSnapped", "headerSnapped", booleanAttribute] }, outputs: { ui5PinButtonToggle: "ui5PinButtonToggle", ui5TitleToggle: "ui5TitleToggle" }, exportAs: ["ui5DynamicPage"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
DynamicPageComponent = __decorate([
    ProxyInputs(['hidePinButton', 'headerPinned', 'showFooter', 'headerSnapped']),
    ProxyOutputs([
        'pin-button-toggle: ui5PinButtonToggle',
        'title-toggle: ui5TitleToggle',
    ]),
    __metadata("design:paramtypes", [])
], DynamicPageComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicPageComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-dynamic-page',
                    template: '<ng-content></ng-content>',
                    inputs: ['hidePinButton', 'headerPinned', 'showFooter', 'headerSnapped'],
                    outputs: ['ui5PinButtonToggle', 'ui5TitleToggle'],
                    exportAs: 'ui5DynamicPage',
                }]
        }], ctorParameters: () => [], propDecorators: { hidePinButton: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], headerPinned: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showFooter: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], headerSnapped: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { DynamicPageComponent };
