// @ui5/webcomponents-ngx/esm2022/main/tokenizer/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Tokenizer.js';
import * as i0 from "@angular/core";
let TokenizerComponent = class TokenizerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TokenizerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TokenizerComponent, isStandalone: true, selector: "ui5-tokenizer", inputs: { readonly: ["readonly", "readonly", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5TokenDelete: "ui5TokenDelete", ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5Tokenizer"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TokenizerComponent = __decorate([
    ProxyInputs(['readonly', 'disabled', 'accessibleName', 'accessibleNameRef']),
    ProxyOutputs([
        'token-delete: ui5TokenDelete',
        'selection-change: ui5SelectionChange',
    ]),
    __metadata("design:paramtypes", [])
], TokenizerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TokenizerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tokenizer',
                    template: '<ng-content></ng-content>',
                    inputs: ['readonly', 'disabled', 'accessibleName', 'accessibleNameRef'],
                    outputs: ['ui5TokenDelete', 'ui5SelectionChange'],
                    exportAs: 'ui5Tokenizer',
                }]
        }], ctorParameters: () => [], propDecorators: { readonly: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { TokenizerComponent };
