// @ui5/webcomponents-ngx/esm2022/main/busy-indicator/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/BusyIndicator.js';
import * as i0 from "@angular/core";
let BusyIndicatorComponent = class BusyIndicatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BusyIndicatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: BusyIndicatorComponent, isStandalone: true, selector: "ui5-busy-indicator", inputs: { text: "text", size: "size", active: ["active", "active", booleanAttribute], delay: "delay", textPlacement: "textPlacement" }, exportAs: ["ui5BusyIndicator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
BusyIndicatorComponent = __decorate([
    ProxyInputs(['text', 'size', 'active', 'delay', 'textPlacement']),
    __metadata("design:paramtypes", [])
], BusyIndicatorComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BusyIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-busy-indicator',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'size', 'active', 'delay', 'textPlacement'],
                    exportAs: 'ui5BusyIndicator',
                }]
        }], ctorParameters: () => [], propDecorators: { active: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { BusyIndicatorComponent };
