// @ui5/webcomponents-ngx/esm2022/theming/base-webcomponents-theming-provider.mjs
import { inject, Injectable } from "@angular/core";
import { Ui5ThemingService } from "@ui5/theming-ngx";
import { BehaviorSubject, map, of, switchMap } from "rxjs";
import supportedThemes from './supported-themes';
import { setTheme } from '@ui5/webcomponents-base/dist/config/Theme.js';
import * as i0 from "@angular/core";
export class WebcomponentsThemingProvider {
    /** @hidden */
    constructor(registerThemes) {
        this.registerThemes = registerThemes;
        this.availableThemes$ = new BehaviorSubject(supportedThemes);
        this._globalThemingService = inject(Ui5ThemingService, { optional: true });
        registerThemes().then(() => {
            this._globalThemingService?.registerProvider(this);
        });
    }
    /** @hidden */
    ngOnDestroy() {
        this._globalThemingService?.unregisterProvider(this);
    }
    supportsTheme(theme) {
        return this.getAvailableThemes().pipe(map((themes) => themes.includes(theme)));
    }
    getAvailableThemes() {
        return this.availableThemes$.asObservable();
    }
    setTheme(theme) {
        return this.supportsTheme(theme).pipe(switchMap((supported) => {
            if (supported) {
                setTheme(theme);
            }
            return of(supported);
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider, deps: "invalid", target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: undefined }] });
