// @ui5/webcomponents-ngx/esm2022/theming/theming.module.mjs
import { NgModule } from "@angular/core";
import { Ui5WebcomponentsThemingService } from "./theming.service";
import * as i0 from "@angular/core";
import * as i1 from "./theming.service";
export class Ui5WebcomponentsThemingModule {
    constructor(theming) {
        this.theming = theming;
    }
    static forRoot() {
        return {
            ngModule: Ui5WebcomponentsThemingModule,
            providers: [Ui5WebcomponentsThemingService]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule, deps: [{ token: i1.Ui5WebcomponentsThemingService }], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [],
                    declarations: []
                }]
        }], ctorParameters: () => [{ type: i1.Ui5WebcomponentsThemingService }] });
