// @ui5/webcomponents-ngx/schematics/add-i18n/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addI18n = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ts = __importStar(require("typescript"));
const workspace_1 = require("@schematics/angular/utility/workspace");
const change_1 = require("@schematics/angular/utility/change");
const ng_ast_utils_1 = require("@schematics/angular/utility/ng-ast-utils");
const ast_utils_1 = require("@schematics/angular/utility/ast-utils");
const standalone_1 = require("@schematics/angular/private/standalone");
const get_project_definition_1 = require("../utils/get-project-definition");
const project_main_file_1 = require("../utils/project-main-file");
const getModuleDeclaration_1 = require("../utils/getModuleDeclaration");
const getSourceFile_1 = require("../utils/getSourceFile");
const find_providers_literal_1 = require("../utils/find-providers-literal");
function addI18n(options) {
    return (tree, context) => __awaiter(this, void 0, void 0, function* () {
        return (0, workspace_1.updateWorkspace)((workspace) => __awaiter(this, void 0, void 0, function* () {
            if (!options.useI18n) {
                return;
            }
            const projectDefinition = (0, get_project_definition_1.getProjectDefinition)(workspace, options.project);
            const update = addI18nModule(tree, projectDefinition, context, options);
            const exportRecorder = tree.beginUpdate(update.file);
            for (const change of update.changes) {
                if (change instanceof change_1.InsertChange) {
                    exportRecorder.insertLeft(change.pos, change.toAdd);
                }
            }
            tree.commitUpdate(exportRecorder);
        }));
    });
}
exports.addI18n = addI18n;
function addI18nModule(tree, projectDefinition, context, options) {
    var _a;
    try {
        return addModuleToNonStandaloneApp(tree, projectDefinition, context, options);
    }
    catch (e) {
        if ((_a = e.message) === null || _a === void 0 ? void 0 : _a.includes('Bootstrap call not found')) {
            return addModuleToStandaloneApp(tree, projectDefinition, context, options);
        }
        else {
            throw e;
        }
    }
}
function rootProviderConfig(options) {
    return `{
      language: ${JSON.stringify(options.defaultLanguage)},
      fetchDefaultLanguage: true,
      bundle: {
        name: 'app-root-i18n-bundle',
        translations: {
            // You can add your translations here. For more information please refer to the documentation.
            [${JSON.stringify(options.defaultLanguage)}]: {
                'app-root-title': 'Hello World!'
            }
        }
      }
    }`;
}
function addModuleToNonStandaloneApp(tree, projectDefinition, context, options) {
    const appModulePath = (0, ng_ast_utils_1.getAppModulePath)(tree, (0, project_main_file_1.getProjectMainFile)(projectDefinition));
    if (!appModulePath) {
        throw new schematics_1.SchematicsException('Could not find root module for defined project.');
    }
    const appModuleSource = (0, getSourceFile_1.getSourceFile)(tree, appModulePath);
    const i18nModule = (0, getModuleDeclaration_1.getModuleDeclaration)(appModuleSource, 'Ui5I18nModule');
    const moduleConfig = `\nUi5I18nModule.forRoot(${rootProviderConfig(options)})`;
    if (i18nModule) {
        i18nModule.getChildren(appModuleSource);
        const i18nModuleDecl = i18nModule.getFullText();
        const appModuleContent = tree.readText(appModulePath).split(i18nModuleDecl).join(moduleConfig);
        tree.overwrite(appModulePath, appModuleContent);
        context.logger.info('Found previous Ui5I18nModule. Replaced with new one.');
        return { changes: [], file: appModulePath };
    }
    const changes = [
        ...(0, ast_utils_1.addImportToModule)(appModuleSource, appModulePath, moduleConfig, null),
        (0, ast_utils_1.insertImport)(appModuleSource, appModulePath, 'Ui5I18nModule', '@ui5/webcomponents-ngx/i18n')
    ];
    return { changes, file: appModulePath };
}
function addModuleToStandaloneApp(tree, projectDefinition, context, options) {
    const mainFile = (0, project_main_file_1.getProjectMainFile)(projectDefinition);
    const mainFileSource = (0, getSourceFile_1.getSourceFile)(tree, mainFile);
    const bootstrapCall = (0, standalone_1.findBootstrapApplicationCall)(mainFileSource);
    if (!bootstrapCall) {
        throw new schematics_1.SchematicsException('Could not find bootstrap call in main.ts');
    }
    const providersLiteral = (0, find_providers_literal_1.findProvidersLiteral)(bootstrapCall);
    const recorder = tree.beginUpdate(mainFile.toString());
    const printer = ts.createPrinter();
    const newProviderCall = ts.factory.createCallExpression(ts.factory.createIdentifier('i18nRootProviders'), undefined, [ts.factory.createObjectLiteralExpression([
            ts.factory.createPropertyAssignment('language', ts.factory.createStringLiteral(options.defaultLanguage)),
            ts.factory.createPropertyAssignment('fetchDefaultLanguage', ts.factory.createTrue()),
            ts.factory.createPropertyAssignment('bundle', ts.factory.createObjectLiteralExpression([
                ts.factory.createPropertyAssignment('name', ts.factory.createStringLiteral('app-root-i18n-bundle')),
                ts.factory.createPropertyAssignment('translations', ts.factory.createObjectLiteralExpression([
                    ts.factory.createPropertyAssignment(ts.factory.createStringLiteral(options.defaultLanguage), ts.factory.createObjectLiteralExpression([
                        ts.factory.createPropertyAssignment(ts.factory.createStringLiteral('app-root-title'), ts.factory.createStringLiteral('Hello World!')),
                    ])),
                ])),
            ])),
        ], true)]);
    if (!providersLiteral) {
        const bootstrapCallParams = [...bootstrapCall.arguments];
        bootstrapCallParams[1] = ts.factory.createObjectLiteralExpression([createProvidersAssignment([newProviderCall])], true);
        const newCall = ts.factory.updateCallExpression(bootstrapCall, bootstrapCall.expression, bootstrapCall.typeArguments, bootstrapCallParams);
        recorder.remove(bootstrapCall.getStart(), bootstrapCall.getWidth());
        recorder.insertRight(bootstrapCall.getStart(), printer.printNode(ts.EmitHint.Unspecified, newCall, mainFileSource));
    }
    else {
        const existingUsageIndex = providersLiteral.elements.findIndex(el => ts.isCallExpression(el) && ts.isIdentifier(el.expression) && el.expression.escapedText === 'i18nRootProviders');
        const newProvidersLiteral = ts.factory.createArrayLiteralExpression(providersLiteral.elements.filter((_el, i) => i !== existingUsageIndex).concat(newProviderCall));
        if (existingUsageIndex !== -1) {
            context.logger.info('Found previous i18nRootProviders call. Replaced with new one.');
        }
        recorder.remove(providersLiteral.getStart(), providersLiteral.getWidth());
        recorder.insertRight(providersLiteral.getStart(), printer.printNode(ts.EmitHint.Unspecified, newProvidersLiteral, mainFileSource));
    }
    const importChange = (0, ast_utils_1.insertImport)(mainFileSource, mainFile, 'i18nRootProviders', '@ui5/webcomponents-ngx/i18n');
    if (importChange instanceof change_1.InsertChange) {
        recorder.insertLeft(importChange.pos, importChange.toAdd);
    }
    tree.commitUpdate(recorder);
    return { changes: [], file: mainFile };
}
function createProvidersAssignment(elements = []) {
    return ts.factory.createPropertyAssignment('providers', ts.factory.createArrayLiteralExpression(elements));
}
