// @ui5/webcomponents-ngx/esm2022/main/progress-indicator/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ProgressIndicator.js';
import * as i0 from "@angular/core";
let ProgressIndicatorComponent = class ProgressIndicatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProgressIndicatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ProgressIndicatorComponent, isStandalone: true, selector: "ui5-progress-indicator", inputs: { accessibleName: "accessibleName", hideValue: ["hideValue", "hideValue", booleanAttribute], value: "value", displayValue: "displayValue", valueState: "valueState" }, exportAs: ["ui5ProgressIndicator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ProgressIndicatorComponent = __decorate([
    ProxyInputs([
        'accessibleName',
        'hideValue',
        'value',
        'displayValue',
        'valueState',
    ]),
    __metadata("design:paramtypes", [])
], ProgressIndicatorComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProgressIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-progress-indicator',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accessibleName',
                        'hideValue',
                        'value',
                        'displayValue',
                        'valueState',
                    ],
                    exportAs: 'ui5ProgressIndicator',
                }]
        }], ctorParameters: () => [], propDecorators: { hideValue: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ProgressIndicatorComponent };
