// @ui5/webcomponents-ngx/esm2022/fiori/timeline-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/TimelineItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let TimelineItemComponent = class TimelineItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TimelineItemComponent, isStandalone: true, selector: "ui5-timeline-item", inputs: { icon: "icon", name: "name", nameClickable: ["nameClickable", "nameClickable", booleanAttribute], titleText: "titleText", subtitleText: "subtitleText" }, outputs: { ui5NameClick: "ui5NameClick" }, exportAs: ["ui5TimelineItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TimelineItemComponent = __decorate([
    ProxyInputs(['icon', 'name', 'nameClickable', 'titleText', 'subtitleText']),
    ProxyOutputs(['name-click: ui5NameClick']),
    __metadata("design:paramtypes", [])
], TimelineItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-timeline-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['icon', 'name', 'nameClickable', 'titleText', 'subtitleText'],
                    outputs: ['ui5NameClick'],
                    exportAs: 'ui5TimelineItem',
                }]
        }], ctorParameters: () => [], propDecorators: { nameClickable: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { TimelineItemComponent };
