// @ui5/webcomponents-ngx/esm2022/main/toolbar-select-option/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarSelectOption.js';
import * as i0 from "@angular/core";
let ToolbarSelectOptionComponent = class ToolbarSelectOptionComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSelectOptionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarSelectOptionComponent, isStandalone: true, selector: "ui5-toolbar-select-option", inputs: { selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5ToolbarSelectOption"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarSelectOptionComponent = __decorate([
    ProxyInputs(['selected']),
    __metadata("design:paramtypes", [])
], ToolbarSelectOptionComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSelectOptionComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-select-option',
                    template: '<ng-content></ng-content>',
                    inputs: ['selected'],
                    exportAs: 'ui5ToolbarSelectOption',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { ToolbarSelectOptionComponent };
