// @ui5/webcomponents-ngx/esm2022/fiori/notification-list-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/NotificationListItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let NotificationListItemComponent = class NotificationListItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: NotificationListItemComponent, isStandalone: true, selector: "ui5-li-notification", inputs: { titleText: "titleText", read: ["read", "read", booleanAttribute], loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay", wrappingType: "wrappingType", state: "state", showClose: ["showClose", "showClose", booleanAttribute], importance: "importance" }, outputs: { ui5Close: "ui5Close" }, exportAs: ["ui5LiNotification"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
NotificationListItemComponent = __decorate([
    ProxyInputs([
        'titleText',
        'read',
        'loading',
        'loadingDelay',
        'wrappingType',
        'state',
        'showClose',
        'importance',
    ]),
    ProxyOutputs(['close: ui5Close']),
    __metadata("design:paramtypes", [])
], NotificationListItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-li-notification',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'titleText',
                        'read',
                        'loading',
                        'loadingDelay',
                        'wrappingType',
                        'state',
                        'showClose',
                        'importance',
                    ],
                    outputs: ['ui5Close'],
                    exportAs: 'ui5LiNotification',
                }]
        }], ctorParameters: () => [], propDecorators: { read: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], loading: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], showClose: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { NotificationListItemComponent };
