// @ui5/webcomponents-ngx/esm2022/fiori/media-gallery-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/MediaGalleryItem.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let MediaGalleryItemComponent = class MediaGalleryItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MediaGalleryItemComponent, isStandalone: true, selector: "ui5-media-gallery-item", inputs: { selected: ["selected", "selected", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], layout: "layout" }, exportAs: ["ui5MediaGalleryItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MediaGalleryItemComponent = __decorate([
    ProxyInputs(['selected', 'disabled', 'layout']),
    __metadata("design:paramtypes", [])
], MediaGalleryItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-media-gallery-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['selected', 'disabled', 'layout'],
                    exportAs: 'ui5MediaGalleryItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { MediaGalleryItemComponent };
