// @ui5/webcomponents-ngx/esm2022/config/index.mjs
import { Inject, NgModule } from '@angular/core';
import { setAnimationMode } from '@ui5/webcomponents-base/dist/config/AnimationMode.js';
import * as i0 from "@angular/core";
const setters = {
    animationMode: setAnimationMode,
};
export class Ui5WebcomponentsConfigModule {
    constructor(config) {
        console.log({ config });
    }
    static forRoot(config) {
        return {
            ngModule: Ui5WebcomponentsConfigModule,
            providers: [
                {
                    provide: 'rootConfig',
                    useFactory: () => {
                        Object.entries(config).filter(([, val]) => val !== undefined).forEach(([key, val]) => {
                            setters[key](val);
                        });
                        return config;
                    }
                }
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsConfigModule, deps: [{ token: 'rootConfig' }], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsConfigModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsConfigModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsConfigModule, decorators: [{
            type: NgModule,
            args: [{}]
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: ['rootConfig']
                }] }] });
