// @ui5/webcomponents-ngx/esm2022/main/form-item/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, NgZone, inject, } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/FormItem.js';
import * as i0 from "@angular/core";
let FormItemComponent = class FormItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: FormItemComponent, isStandalone: true, selector: "ui5-form-item", inputs: { columnSpan: "columnSpan" }, exportAs: ["ui5FormItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FormItemComponent = __decorate([
    ProxyInputs(['columnSpan']),
    __metadata("design:paramtypes", [])
], FormItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-form-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['columnSpan'],
                    exportAs: 'ui5FormItem',
                }]
        }], ctorParameters: () => [] });
export { FormItemComponent };
