// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-product-switch-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/ProductSwitchItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let ProductSwitchItemComponent = class ProductSwitchItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProductSwitchItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ProductSwitchItemComponent, isStandalone: true, selector: "ui5-product-switch-item", inputs: { titleText: "titleText", subtitleText: "subtitleText", icon: "icon", target: "target", targetSrc: "targetSrc" }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5ProductSwitchItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ProductSwitchItemComponent = __decorate([
    ProxyInputs(['titleText', 'subtitleText', 'icon', 'target', 'targetSrc']),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], ProductSwitchItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProductSwitchItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-product-switch-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['titleText', 'subtitleText', 'icon', 'target', 'targetSrc'],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5ProductSwitchItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ProductSwitchItemComponent };
