// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-theming.mjs
import * as i0 from '@angular/core';
import { inject, Injectable, NgModule } from '@angular/core';
import { Ui5ThemingService } from '@ui5/theming-ngx';
import { BehaviorSubject, map, of, switchMap } from 'rxjs';
import { setTheme } from '@ui5/webcomponents-base/dist/config/Theme.js';

var supportedThemes = ["node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_fiori_3/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_fiori_3_dark/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_fiori_3_hcb/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_fiori_3_hcw/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_dark/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_dark_exp/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_exp/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_hcb/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_hcb_exp/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_hcw/parameters-bundle.css.json", "node_modules/@ui5/webcomponents-theming/dist/generated/assets/themes/sap_horizon_hcw_exp/parameters-bundle.css.json"];

class WebcomponentsThemingProvider {
    /** @hidden */
    constructor(registerThemes) {
        this.registerThemes = registerThemes;
        this.availableThemes$ = new BehaviorSubject(supportedThemes);
        this._globalThemingService = inject(Ui5ThemingService, { optional: true });
        registerThemes().then(() => {
            this._globalThemingService?.registerProvider(this);
        });
    }
    /** @hidden */
    ngOnDestroy() {
        this._globalThemingService?.unregisterProvider(this);
    }
    supportsTheme(theme) {
        return this.getAvailableThemes().pipe(map((themes) => themes.includes(theme)));
    }
    getAvailableThemes() {
        return this.availableThemes$.asObservable();
    }
    setTheme(theme) {
        return this.supportsTheme(theme).pipe(switchMap((supported) => {
            if (supported) {
                setTheme(theme);
            }
            return of(supported);
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider, deps: "invalid", target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WebcomponentsThemingProvider, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: undefined }] });

/**
 * Theming service specifically for the ui5/webcomponents-ngx components.
 */
class Ui5WebcomponentsThemingService extends WebcomponentsThemingProvider {
    /** @hidden */
    constructor() {
        super(() => import('@ui5/webcomponents-theming/dist/generated/json-imports/Themes.js'));
        this.name = 'ui5-webcomponents-theming-service';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });

class Ui5WebcomponentsThemingModule {
    constructor(theming) {
        this.theming = theming;
    }
    static forRoot() {
        return {
            ngModule: Ui5WebcomponentsThemingModule,
            providers: [Ui5WebcomponentsThemingService]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule, deps: [{ token: Ui5WebcomponentsThemingService }], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5WebcomponentsThemingModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [],
                    declarations: []
                }]
        }], ctorParameters: () => [{ type: Ui5WebcomponentsThemingService }] });

/**
 * Generated bundle index. Do not edit.
 */

export { Ui5WebcomponentsThemingModule, WebcomponentsThemingProvider };
