// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-menu-separator.mjs
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents/dist/MenuSeparator.js';

class MenuSeparatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuSeparatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: MenuSeparatorComponent, isStandalone: true, selector: "ui5-menu-separator", exportAs: ["ui5MenuSeparator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuSeparatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-menu-separator',
                    template: '<ng-content></ng-content>',
                    exportAs: 'ui5MenuSeparator',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { MenuSeparatorComponent };
