// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-view-settings-dialog.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/ViewSettingsDialog.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let ViewSettingsDialogComponent = class ViewSettingsDialogComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ViewSettingsDialogComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ViewSettingsDialogComponent, isStandalone: true, selector: "ui5-view-settings-dialog", inputs: { sortDescending: ["sortDescending", "sortDescending", booleanAttribute], open: ["open", "open", booleanAttribute] }, outputs: { ui5Confirm: "ui5Confirm", ui5Cancel: "ui5Cancel", ui5BeforeOpen: "ui5BeforeOpen", ui5Open: "ui5Open", ui5Close: "ui5Close" }, exportAs: ["ui5ViewSettingsDialog"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ViewSettingsDialogComponent = __decorate([
    ProxyInputs(['sortDescending', 'open']),
    ProxyOutputs([
        'confirm: ui5Confirm',
        'cancel: ui5Cancel',
        'before-open: ui5BeforeOpen',
        'open: ui5Open',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], ViewSettingsDialogComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ViewSettingsDialogComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-view-settings-dialog',
                    template: '<ng-content></ng-content>',
                    inputs: ['sortDescending', 'open'],
                    outputs: ['ui5Confirm', 'ui5Cancel', 'ui5BeforeOpen', 'ui5Open', 'ui5Close'],
                    exportAs: 'ui5ViewSettingsDialog',
                }]
        }], ctorParameters: () => [], propDecorators: { sortDescending: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ViewSettingsDialogComponent };
