// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-tree-item-custom.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TreeItemCustom.js';

let TreeItemCustomComponent = class TreeItemCustomComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TreeItemCustomComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TreeItemCustomComponent, isStandalone: true, selector: "ui5-tree-item-custom", inputs: { type: "type", accessibilityAttributes: "accessibilityAttributes", navigated: ["navigated", "navigated", booleanAttribute], tooltip: "tooltip", highlight: "highlight", selected: ["selected", "selected", booleanAttribute], icon: "icon", expanded: ["expanded", "expanded", booleanAttribute], movable: ["movable", "movable", booleanAttribute], indeterminate: ["indeterminate", "indeterminate", booleanAttribute], hasChildren: ["hasChildren", "hasChildren", booleanAttribute], additionalTextState: "additionalTextState", accessibleName: "accessibleName", hideSelectionElement: ["hideSelectionElement", "hideSelectionElement", booleanAttribute] }, outputs: { ui5DetailClick: "ui5DetailClick" }, exportAs: ["ui5TreeItemCustom"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TreeItemCustomComponent = __decorate([
    ProxyInputs([
        'type',
        'accessibilityAttributes',
        'navigated',
        'tooltip',
        'highlight',
        'selected',
        'icon',
        'expanded',
        'movable',
        'indeterminate',
        'hasChildren',
        'additionalTextState',
        'accessibleName',
        'hideSelectionElement',
    ]),
    ProxyOutputs(['detail-click: ui5DetailClick']),
    __metadata("design:paramtypes", [])
], TreeItemCustomComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TreeItemCustomComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tree-item-custom',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'type',
                        'accessibilityAttributes',
                        'navigated',
                        'tooltip',
                        'highlight',
                        'selected',
                        'icon',
                        'expanded',
                        'movable',
                        'indeterminate',
                        'hasChildren',
                        'additionalTextState',
                        'accessibleName',
                        'hideSelectionElement',
                    ],
                    outputs: ['ui5DetailClick'],
                    exportAs: 'ui5TreeItemCustom',
                }]
        }], ctorParameters: () => [], propDecorators: { navigated: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], expanded: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], movable: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], indeterminate: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hasChildren: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hideSelectionElement: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TreeItemCustomComponent };
