// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-avatar-group.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/AvatarGroup.js';

let AvatarGroupComponent = class AvatarGroupComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: AvatarGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: AvatarGroupComponent, isStandalone: true, selector: "ui5-avatar-group", inputs: { type: "type", accessibilityAttributes: "accessibilityAttributes" }, outputs: { ui5Click: "ui5Click", ui5Overflow: "ui5Overflow" }, exportAs: ["ui5AvatarGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
AvatarGroupComponent = __decorate([
    ProxyInputs(['type', 'accessibilityAttributes']),
    ProxyOutputs(['click: ui5Click', 'overflow: ui5Overflow']),
    __metadata("design:paramtypes", [])
], AvatarGroupComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: AvatarGroupComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-avatar-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['type', 'accessibilityAttributes'],
                    outputs: ['ui5Click', 'ui5Overflow'],
                    exportAs: 'ui5AvatarGroup',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { AvatarGroupComponent };
