// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-flexible-column-layout.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/FlexibleColumnLayout.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let FlexibleColumnLayoutComponent = class FlexibleColumnLayoutComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FlexibleColumnLayoutComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: FlexibleColumnLayoutComponent, isStandalone: true, selector: "ui5-flexible-column-layout", inputs: { layout: "layout", disableResizing: ["disableResizing", "disableResizing", booleanAttribute], accessibilityAttributes: "accessibilityAttributes" }, outputs: { ui5LayoutChange: "ui5LayoutChange" }, exportAs: ["ui5FlexibleColumnLayout"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FlexibleColumnLayoutComponent = __decorate([
    ProxyInputs(['layout', 'disableResizing', 'accessibilityAttributes']),
    ProxyOutputs(['layout-change: ui5LayoutChange']),
    __metadata("design:paramtypes", [])
], FlexibleColumnLayoutComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FlexibleColumnLayoutComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-flexible-column-layout',
                    template: '<ng-content></ng-content>',
                    inputs: ['layout', 'disableResizing', 'accessibilityAttributes'],
                    outputs: ['ui5LayoutChange'],
                    exportAs: 'ui5FlexibleColumnLayout',
                }]
        }], ctorParameters: () => [], propDecorators: { disableResizing: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { FlexibleColumnLayoutComponent };
