// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-tab-container.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TabContainer.js';

let TabContainerComponent = class TabContainerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabContainerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TabContainerComponent, isStandalone: true, selector: "ui5-tabcontainer", inputs: { collapsed: ["collapsed", "collapsed", booleanAttribute], tabLayout: "tabLayout", overflowMode: "overflowMode", headerBackgroundDesign: "headerBackgroundDesign", contentBackgroundDesign: "contentBackgroundDesign" }, outputs: { ui5TabSelect: "ui5TabSelect", ui5MoveOver: "ui5MoveOver", ui5Move: "ui5Move" }, exportAs: ["ui5Tabcontainer"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TabContainerComponent = __decorate([
    ProxyInputs([
        'collapsed',
        'tabLayout',
        'overflowMode',
        'headerBackgroundDesign',
        'contentBackgroundDesign',
    ]),
    ProxyOutputs([
        'tab-select: ui5TabSelect',
        'move-over: ui5MoveOver',
        'move: ui5Move',
    ]),
    __metadata("design:paramtypes", [])
], TabContainerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabContainerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tabcontainer',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'collapsed',
                        'tabLayout',
                        'overflowMode',
                        'headerBackgroundDesign',
                        'contentBackgroundDesign',
                    ],
                    outputs: ['ui5TabSelect', 'ui5MoveOver', 'ui5Move'],
                    exportAs: 'ui5Tabcontainer',
                }]
        }], ctorParameters: () => [], propDecorators: { collapsed: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TabContainerComponent };
