// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-side-navigation-group.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/SideNavigationGroup.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';

let SideNavigationGroupComponent = class SideNavigationGroupComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SideNavigationGroupComponent, isStandalone: true, selector: "ui5-side-navigation-group", inputs: { text: "text", disabled: ["disabled", "disabled", booleanAttribute], tooltip: "tooltip", expanded: ["expanded", "expanded", booleanAttribute] }, exportAs: ["ui5SideNavigationGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SideNavigationGroupComponent = __decorate([
    ProxyInputs(['text', 'disabled', 'tooltip', 'expanded']),
    __metadata("design:paramtypes", [])
], SideNavigationGroupComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationGroupComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-side-navigation-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'disabled', 'tooltip', 'expanded'],
                    exportAs: 'ui5SideNavigationGroup',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], expanded: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SideNavigationGroupComponent };
