// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-filter-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/FilterItem.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';

let FilterItemComponent = class FilterItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FilterItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: FilterItemComponent, isStandalone: true, selector: "ui5-filter-item", inputs: { text: "text", additionalText: "additionalText" }, exportAs: ["ui5FilterItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FilterItemComponent = __decorate([
    ProxyInputs(['text', 'additionalText']),
    __metadata("design:paramtypes", [])
], FilterItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FilterItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-filter-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'additionalText'],
                    exportAs: 'ui5FilterItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { FilterItemComponent };
