// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-file-uploader.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/FileUploader.js';

let FileUploaderComponent = class FileUploaderComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FileUploaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: FileUploaderComponent, isStandalone: true, selector: "ui5-file-uploader", inputs: { accept: "accept", hideInput: ["hideInput", "hideInput", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], multiple: ["multiple", "multiple", booleanAttribute], name: "name", placeholder: "placeholder", value: "value", valueState: "valueState" }, outputs: { ui5Change: "ui5Change" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5FileUploader"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FileUploaderComponent = __decorate([
    ProxyInputs([
        'accept',
        'hideInput',
        'disabled',
        'multiple',
        'name',
        'placeholder',
        'value',
        'valueState',
    ]),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], FileUploaderComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FileUploaderComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-file-uploader',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accept',
                        'hideInput',
                        'disabled',
                        'multiple',
                        'name',
                        'placeholder',
                        'value',
                        'valueState',
                    ],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5FileUploader',
                    hostDirectives: [GenericControlValueAccessor],
                    host: { '(change)': '_cva?.onChange?.(cvaValue);' },
                }]
        }], ctorParameters: () => [], propDecorators: { hideInput: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], multiple: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { FileUploaderComponent };
