// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-multi-combo-box-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/MultiComboBoxItem.js';

let MultiComboBoxItemComponent = class MultiComboBoxItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiComboBoxItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MultiComboBoxItemComponent, isStandalone: true, selector: "ui5-mcb-item", inputs: { text: "text", additionalText: "additionalText", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5McbItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MultiComboBoxItemComponent = __decorate([
    ProxyInputs(['text', 'additionalText', 'selected']),
    __metadata("design:paramtypes", [])
], MultiComboBoxItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MultiComboBoxItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-mcb-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'additionalText', 'selected'],
                    exportAs: 'ui5McbItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { MultiComboBoxItemComponent };
