// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-calendar-legend-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CalendarLegendItem.js';

let CalendarLegendItemComponent = class CalendarLegendItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarLegendItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: CalendarLegendItemComponent, isStandalone: true, selector: "ui5-calendar-legend-item", inputs: { text: "text", type: "type" }, exportAs: ["ui5CalendarLegendItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CalendarLegendItemComponent = __decorate([
    ProxyInputs(['text', 'type']),
    __metadata("design:paramtypes", [])
], CalendarLegendItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarLegendItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-calendar-legend-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'type'],
                    exportAs: 'ui5CalendarLegendItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { CalendarLegendItemComponent };
