// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-date-picker.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/DatePicker.js';

let DatePickerComponent = class DatePickerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DatePickerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: DatePickerComponent, isStandalone: true, selector: "ui5-date-picker", inputs: { primaryCalendarType: "primaryCalendarType", secondaryCalendarType: "secondaryCalendarType", formatPattern: "formatPattern", minDate: "minDate", maxDate: "maxDate", value: "value", valueState: "valueState", required: ["required", "required", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], placeholder: "placeholder", name: "name", hideWeekNumbers: ["hideWeekNumbers", "hideWeekNumbers", booleanAttribute], open: ["open", "open", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input", ui5ValueStateChange: "ui5ValueStateChange" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5DatePicker"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
DatePickerComponent = __decorate([
    ProxyInputs([
        'primaryCalendarType',
        'secondaryCalendarType',
        'formatPattern',
        'minDate',
        'maxDate',
        'value',
        'valueState',
        'required',
        'disabled',
        'readonly',
        'placeholder',
        'name',
        'hideWeekNumbers',
        'open',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'input: ui5Input',
        'value-state-change: ui5ValueStateChange',
    ]),
    __metadata("design:paramtypes", [])
], DatePickerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DatePickerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-date-picker',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'primaryCalendarType',
                        'secondaryCalendarType',
                        'formatPattern',
                        'minDate',
                        'maxDate',
                        'value',
                        'valueState',
                        'required',
                        'disabled',
                        'readonly',
                        'placeholder',
                        'name',
                        'hideWeekNumbers',
                        'open',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: ['ui5Change', 'ui5Input', 'ui5ValueStateChange'],
                    exportAs: 'ui5DatePicker',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hideWeekNumbers: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { DatePickerComponent };
