// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-text-area.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TextArea.js';

let TextAreaComponent = class TextAreaComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TextAreaComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TextAreaComponent, isStandalone: true, selector: "ui5-textarea", inputs: { value: "value", disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], required: ["required", "required", booleanAttribute], placeholder: "placeholder", valueState: "valueState", rows: "rows", maxlength: "maxlength", showExceededText: ["showExceededText", "showExceededText", booleanAttribute], growing: ["growing", "growing", booleanAttribute], growingMaxRows: "growingMaxRows", name: "name", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input", ui5Select: "ui5Select", ui5Scroll: "ui5Scroll" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5Textarea"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TextAreaComponent = __decorate([
    ProxyInputs([
        'value',
        'disabled',
        'readonly',
        'required',
        'placeholder',
        'valueState',
        'rows',
        'maxlength',
        'showExceededText',
        'growing',
        'growingMaxRows',
        'name',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'input: ui5Input',
        'select: ui5Select',
        'scroll: ui5Scroll',
    ]),
    __metadata("design:paramtypes", [])
], TextAreaComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TextAreaComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-textarea',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'value',
                        'disabled',
                        'readonly',
                        'required',
                        'placeholder',
                        'valueState',
                        'rows',
                        'maxlength',
                        'showExceededText',
                        'growing',
                        'growingMaxRows',
                        'name',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: ['ui5Change', 'ui5Input', 'ui5Select', 'ui5Scroll'],
                    exportAs: 'ui5Textarea',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], showExceededText: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], growing: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TextAreaComponent };
