// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-breadcrumbs.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Breadcrumbs.js';

let BreadcrumbsComponent = class BreadcrumbsComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BreadcrumbsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: BreadcrumbsComponent, isStandalone: true, selector: "ui5-breadcrumbs", inputs: { design: "design", separators: "separators" }, outputs: { ui5ItemClick: "ui5ItemClick" }, exportAs: ["ui5Breadcrumbs"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
BreadcrumbsComponent = __decorate([
    ProxyInputs(['design', 'separators']),
    ProxyOutputs(['item-click: ui5ItemClick']),
    __metadata("design:paramtypes", [])
], BreadcrumbsComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BreadcrumbsComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-breadcrumbs',
                    template: '<ng-content></ng-content>',
                    inputs: ['design', 'separators'],
                    outputs: ['ui5ItemClick'],
                    exportAs: 'ui5Breadcrumbs',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { BreadcrumbsComponent };
