// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-toggle-button.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToggleButton.js';

let ToggleButtonComponent = class ToggleButtonComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToggleButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToggleButtonComponent, isStandalone: true, selector: "ui5-toggle-button", inputs: { design: "design", disabled: ["disabled", "disabled", booleanAttribute], icon: "icon", endIcon: "endIcon", submits: ["submits", "submits", booleanAttribute], tooltip: "tooltip", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibilityAttributes: "accessibilityAttributes", type: "type", accessibleRole: "accessibleRole", pressed: ["pressed", "pressed", booleanAttribute] }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5ToggleButton"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToggleButtonComponent = __decorate([
    ProxyInputs([
        'design',
        'disabled',
        'icon',
        'endIcon',
        'submits',
        'tooltip',
        'accessibleName',
        'accessibleNameRef',
        'accessibilityAttributes',
        'type',
        'accessibleRole',
        'pressed',
    ]),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], ToggleButtonComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToggleButtonComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toggle-button',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'design',
                        'disabled',
                        'icon',
                        'endIcon',
                        'submits',
                        'tooltip',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibilityAttributes',
                        'type',
                        'accessibleRole',
                        'pressed',
                    ],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5ToggleButton',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], submits: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], pressed: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ToggleButtonComponent };
