// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-toolbar-spacer.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarSpacer.js';

let ToolbarSpacerComponent = class ToolbarSpacerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSpacerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarSpacerComponent, isStandalone: true, selector: "ui5-toolbar-spacer", inputs: { overflowPriority: "overflowPriority", preventOverflowClosing: ["preventOverflowClosing", "preventOverflowClosing", booleanAttribute], width: "width" }, exportAs: ["ui5ToolbarSpacer"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarSpacerComponent = __decorate([
    ProxyInputs(['overflowPriority', 'preventOverflowClosing', 'width']),
    __metadata("design:paramtypes", [])
], ToolbarSpacerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSpacerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-spacer',
                    template: '<ng-content></ng-content>',
                    inputs: ['overflowPriority', 'preventOverflowClosing', 'width'],
                    exportAs: 'ui5ToolbarSpacer',
                }]
        }], ctorParameters: () => [], propDecorators: { preventOverflowClosing: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ToolbarSpacerComponent };
