// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-text.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Text.js';

let TextComponent = class TextComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TextComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TextComponent, isStandalone: true, selector: "ui5-text", inputs: { maxLines: "maxLines" }, exportAs: ["ui5Text"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TextComponent = __decorate([
    ProxyInputs(['maxLines']),
    __metadata("design:paramtypes", [])
], TextComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TextComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-text',
                    template: '<ng-content></ng-content>',
                    inputs: ['maxLines'],
                    exportAs: 'ui5Text',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { TextComponent };
