// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-color-palette.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ColorPalette.js';

let ColorPaletteComponent = class ColorPaletteComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ColorPaletteComponent, isStandalone: true, selector: "ui5-color-palette", outputs: { ui5ItemClick: "ui5ItemClick" }, exportAs: ["ui5ColorPalette"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ColorPaletteComponent = __decorate([
    ProxyOutputs(['item-click: ui5ItemClick']),
    __metadata("design:paramtypes", [])
], ColorPaletteComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-color-palette',
                    template: '<ng-content></ng-content>',
                    outputs: ['ui5ItemClick'],
                    exportAs: 'ui5ColorPalette',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ColorPaletteComponent };
