// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-rating-indicator.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/RatingIndicator.js';

let RatingIndicatorComponent = class RatingIndicatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: RatingIndicatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: RatingIndicatorComponent, isStandalone: true, selector: "ui5-rating-indicator", inputs: { value: "value", max: "max", disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", required: ["required", "required", booleanAttribute], tooltip: "tooltip" }, outputs: { ui5Change: "ui5Change" }, exportAs: ["ui5RatingIndicator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
RatingIndicatorComponent = __decorate([
    ProxyInputs([
        'value',
        'max',
        'disabled',
        'readonly',
        'accessibleName',
        'accessibleNameRef',
        'required',
        'tooltip',
    ]),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], RatingIndicatorComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: RatingIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-rating-indicator',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'value',
                        'max',
                        'disabled',
                        'readonly',
                        'accessibleName',
                        'accessibleNameRef',
                        'required',
                        'tooltip',
                    ],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5RatingIndicator',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { RatingIndicatorComponent };
