// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-color-palette-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ColorPaletteItem.js';

let ColorPaletteItemComponent = class ColorPaletteItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ColorPaletteItemComponent, isStandalone: true, selector: "ui5-color-palette-item", inputs: { value: "value", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5ColorPaletteItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ColorPaletteItemComponent = __decorate([
    ProxyInputs(['value', 'selected']),
    __metadata("design:paramtypes", [])
], ColorPaletteItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPaletteItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-color-palette-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['value', 'selected'],
                    exportAs: 'ui5ColorPaletteItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ColorPaletteItemComponent };
