// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-suggestion-item-custom.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SuggestionItemCustom.js';

let SuggestionItemCustomComponent = class SuggestionItemCustomComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SuggestionItemCustomComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: SuggestionItemCustomComponent, isStandalone: true, selector: "ui5-suggestion-item-custom", inputs: { text: "text" }, exportAs: ["ui5SuggestionItemCustom"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SuggestionItemCustomComponent = __decorate([
    ProxyInputs(['text']),
    __metadata("design:paramtypes", [])
], SuggestionItemCustomComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SuggestionItemCustomComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-suggestion-item-custom',
                    template: '<ng-content></ng-content>',
                    inputs: ['text'],
                    exportAs: 'ui5SuggestionItemCustom',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { SuggestionItemCustomComponent };
