// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-table-selection.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TableSelection.js';

let TableSelectionComponent = class TableSelectionComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableSelectionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TableSelectionComponent, isStandalone: true, selector: "ui5-table-selection", inputs: { mode: "mode", selected: "selected" }, outputs: { ui5Change: "ui5Change" }, exportAs: ["ui5TableSelection"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableSelectionComponent = __decorate([
    ProxyInputs(['mode', 'selected']),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], TableSelectionComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableSelectionComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table-selection',
                    template: '<ng-content></ng-content>',
                    inputs: ['mode', 'selected'],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5TableSelection',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { TableSelectionComponent };
