// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-side-navigation.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/SideNavigation.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let SideNavigationComponent = class SideNavigationComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SideNavigationComponent, isStandalone: true, selector: "ui5-side-navigation", inputs: { collapsed: ["collapsed", "collapsed", booleanAttribute] }, outputs: { ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5SideNavigation"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SideNavigationComponent = __decorate([
    ProxyInputs(['collapsed']),
    ProxyOutputs(['selection-change: ui5SelectionChange']),
    __metadata("design:paramtypes", [])
], SideNavigationComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SideNavigationComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-side-navigation',
                    template: '<ng-content></ng-content>',
                    inputs: ['collapsed'],
                    outputs: ['ui5SelectionChange'],
                    exportAs: 'ui5SideNavigation',
                }]
        }], ctorParameters: () => [], propDecorators: { collapsed: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SideNavigationComponent };
