// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-generic-cva.mjs
import * as i0 from '@angular/core';
import { Directive } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';

class GenericControlValueAccessor {
    constructor() {
        this.setDisabledState = (isDisabled) => {
            this.host.element['disabled'] = isDisabled;
        };
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    writeValue(val) {
        this.host.cvaValue = val;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: GenericControlValueAccessor, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.0.9", type: GenericControlValueAccessor, isStandalone: true, selector: "[noop]", host: { listeners: { "focusout": "onTouched?.()" } }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: GenericControlValueAccessor,
                multi: true,
            },
        ], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: GenericControlValueAccessor, decorators: [{
            type: Directive,
            args: [{
                    selector: '[noop]',
                    standalone: true,
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: GenericControlValueAccessor,
                            multi: true,
                        },
                    ],
                    host: {
                        '(focusout)': 'onTouched?.()',
                    },
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { GenericControlValueAccessor };
