// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-tree.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Tree.js';

let TreeComponent = class TreeComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TreeComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TreeComponent, isStandalone: true, selector: "ui5-tree", inputs: { selectionMode: "selectionMode", noDataText: "noDataText", headerText: "headerText", footerText: "footerText", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5ItemToggle: "ui5ItemToggle", ui5ItemMouseover: "ui5ItemMouseover", ui5ItemMouseout: "ui5ItemMouseout", ui5ItemClick: "ui5ItemClick", ui5ItemDelete: "ui5ItemDelete", ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5Tree"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TreeComponent = __decorate([
    ProxyInputs([
        'selectionMode',
        'noDataText',
        'headerText',
        'footerText',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs([
        'item-toggle: ui5ItemToggle',
        'item-mouseover: ui5ItemMouseover',
        'item-mouseout: ui5ItemMouseout',
        'item-click: ui5ItemClick',
        'item-delete: ui5ItemDelete',
        'selection-change: ui5SelectionChange',
    ]),
    __metadata("design:paramtypes", [])
], TreeComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TreeComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tree',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'selectionMode',
                        'noDataText',
                        'headerText',
                        'footerText',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: [
                        'ui5ItemToggle',
                        'ui5ItemMouseover',
                        'ui5ItemMouseout',
                        'ui5ItemClick',
                        'ui5ItemDelete',
                        'ui5SelectionChange',
                    ],
                    exportAs: 'ui5Tree',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { TreeComponent };
