// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-segmented-button-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SegmentedButtonItem.js';

let SegmentedButtonItemComponent = class SegmentedButtonItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SegmentedButtonItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SegmentedButtonItemComponent, isStandalone: true, selector: "ui5-segmented-button-item", inputs: { disabled: ["disabled", "disabled", booleanAttribute], selected: ["selected", "selected", booleanAttribute], tooltip: "tooltip", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", icon: "icon" }, exportAs: ["ui5SegmentedButtonItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SegmentedButtonItemComponent = __decorate([
    ProxyInputs([
        'disabled',
        'selected',
        'tooltip',
        'accessibleName',
        'accessibleNameRef',
        'icon',
    ]),
    __metadata("design:paramtypes", [])
], SegmentedButtonItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SegmentedButtonItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-segmented-button-item',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'selected',
                        'tooltip',
                        'accessibleName',
                        'accessibleNameRef',
                        'icon',
                    ],
                    exportAs: 'ui5SegmentedButtonItem',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SegmentedButtonItemComponent };
