// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-segmented-button.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SegmentedButton.js';

let SegmentedButtonComponent = class SegmentedButtonComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SegmentedButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: SegmentedButtonComponent, isStandalone: true, selector: "ui5-segmented-button", inputs: { accessibleName: "accessibleName", selectionMode: "selectionMode" }, outputs: { ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5SegmentedButton"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SegmentedButtonComponent = __decorate([
    ProxyInputs(['accessibleName', 'selectionMode']),
    ProxyOutputs(['selection-change: ui5SelectionChange']),
    __metadata("design:paramtypes", [])
], SegmentedButtonComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SegmentedButtonComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-segmented-button',
                    template: '<ng-content></ng-content>',
                    inputs: ['accessibleName', 'selectionMode'],
                    outputs: ['ui5SelectionChange'],
                    exportAs: 'ui5SegmentedButton',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { SegmentedButtonComponent };
