// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-dynamic-page-header.mjs
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/DynamicPageHeader.js';

class DynamicPageHeaderComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicPageHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: DynamicPageHeaderComponent, isStandalone: true, selector: "ui5-dynamic-page-header", exportAs: ["ui5DynamicPageHeader"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: DynamicPageHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-dynamic-page-header',
                    template: '<ng-content></ng-content>',
                    exportAs: 'ui5DynamicPageHeader',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { DynamicPageHeaderComponent };
