// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-switch.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Switch.js';

let SwitchComponent = class SwitchComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.checked = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.checked;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SwitchComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SwitchComponent, isStandalone: true, selector: "ui5-switch", inputs: { design: "design", checked: ["checked", "checked", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], textOn: "textOn", textOff: "textOff", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", tooltip: "tooltip", required: ["required", "required", booleanAttribute], name: "name" }, outputs: { ui5Change: "ui5Change" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5Switch"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SwitchComponent = __decorate([
    ProxyInputs([
        'design',
        'checked',
        'disabled',
        'textOn',
        'textOff',
        'accessibleName',
        'accessibleNameRef',
        'tooltip',
        'required',
        'name',
    ]),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], SwitchComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SwitchComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-switch',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'design',
                        'checked',
                        'disabled',
                        'textOn',
                        'textOff',
                        'accessibleName',
                        'accessibleNameRef',
                        'tooltip',
                        'required',
                        'name',
                    ],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5Switch',
                    hostDirectives: [GenericControlValueAccessor],
                    host: { '(change)': '_cva?.onChange?.(cvaValue);' },
                }]
        }], ctorParameters: () => [], propDecorators: { checked: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SwitchComponent };
