// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-media-gallery-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/MediaGalleryItem.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';

let MediaGalleryItemComponent = class MediaGalleryItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MediaGalleryItemComponent, isStandalone: true, selector: "ui5-media-gallery-item", inputs: { selected: ["selected", "selected", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], layout: "layout" }, exportAs: ["ui5MediaGalleryItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MediaGalleryItemComponent = __decorate([
    ProxyInputs(['selected', 'disabled', 'layout']),
    __metadata("design:paramtypes", [])
], MediaGalleryItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-media-gallery-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['selected', 'disabled', 'layout'],
                    exportAs: 'ui5MediaGalleryItem',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { MediaGalleryItemComponent };
