// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-card.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Card.js';

let CardComponent = class CardComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: CardComponent, isStandalone: true, selector: "ui5-card", inputs: { accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, exportAs: ["ui5Card"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CardComponent = __decorate([
    ProxyInputs(['accessibleName', 'accessibleNameRef']),
    __metadata("design:paramtypes", [])
], CardComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CardComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-card',
                    template: '<ng-content></ng-content>',
                    inputs: ['accessibleName', 'accessibleNameRef'],
                    exportAs: 'ui5Card',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { CardComponent };
