// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-calendar-date-range.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CalendarDateRange.js';

let CalendarDateRangeComponent = class CalendarDateRangeComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarDateRangeComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: CalendarDateRangeComponent, isStandalone: true, selector: "ui5-date-range", inputs: { startValue: "startValue", endValue: "endValue" }, exportAs: ["ui5DateRange"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CalendarDateRangeComponent = __decorate([
    ProxyInputs(['startValue', 'endValue']),
    __metadata("design:paramtypes", [])
], CalendarDateRangeComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarDateRangeComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-date-range',
                    template: '<ng-content></ng-content>',
                    inputs: ['startValue', 'endValue'],
                    exportAs: 'ui5DateRange',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { CalendarDateRangeComponent };
