// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-menu.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Menu.js';

let MenuComponent = class MenuComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MenuComponent, isStandalone: true, selector: "ui5-menu", inputs: { headerText: "headerText", open: ["open", "open", booleanAttribute], loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay", opener: "opener" }, outputs: { ui5ItemClick: "ui5ItemClick", ui5BeforeOpen: "ui5BeforeOpen", ui5Open: "ui5Open", ui5BeforeClose: "ui5BeforeClose", ui5Close: "ui5Close" }, exportAs: ["ui5Menu"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MenuComponent = __decorate([
    ProxyInputs(['headerText', 'open', 'loading', 'loadingDelay', 'opener']),
    ProxyOutputs([
        'item-click: ui5ItemClick',
        'before-open: ui5BeforeOpen',
        'open: ui5Open',
        'before-close: ui5BeforeClose',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], MenuComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MenuComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-menu',
                    template: '<ng-content></ng-content>',
                    inputs: ['headerText', 'open', 'loading', 'loadingDelay', 'opener'],
                    outputs: [
                        'ui5ItemClick',
                        'ui5BeforeOpen',
                        'ui5Open',
                        'ui5BeforeClose',
                        'ui5Close',
                    ],
                    exportAs: 'ui5Menu',
                }]
        }], ctorParameters: () => [], propDecorators: { open: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], loading: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { MenuComponent };
