// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-timeline-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/TimelineItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let TimelineItemComponent = class TimelineItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TimelineItemComponent, isStandalone: true, selector: "ui5-timeline-item", inputs: { icon: "icon", name: "name", nameClickable: ["nameClickable", "nameClickable", booleanAttribute], titleText: "titleText", subtitleText: "subtitleText" }, outputs: { ui5NameClick: "ui5NameClick" }, exportAs: ["ui5TimelineItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TimelineItemComponent = __decorate([
    ProxyInputs(['icon', 'name', 'nameClickable', 'titleText', 'subtitleText']),
    ProxyOutputs(['name-click: ui5NameClick']),
    __metadata("design:paramtypes", [])
], TimelineItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimelineItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-timeline-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['icon', 'name', 'nameClickable', 'titleText', 'subtitleText'],
                    outputs: ['ui5NameClick'],
                    exportAs: 'ui5TimelineItem',
                }]
        }], ctorParameters: () => [], propDecorators: { nameClickable: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TimelineItemComponent };
