// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-avatar.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Avatar.js';

let AvatarComponent = class AvatarComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: AvatarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: AvatarComponent, isStandalone: true, selector: "ui5-avatar", inputs: { disabled: ["disabled", "disabled", booleanAttribute], interactive: ["interactive", "interactive", booleanAttribute], icon: "icon", fallbackIcon: "fallbackIcon", initials: "initials", shape: "shape", size: "size", colorScheme: "colorScheme", accessibleName: "accessibleName", accessibilityAttributes: "accessibilityAttributes" }, exportAs: ["ui5Avatar"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
AvatarComponent = __decorate([
    ProxyInputs([
        'disabled',
        'interactive',
        'icon',
        'fallbackIcon',
        'initials',
        'shape',
        'size',
        'colorScheme',
        'accessibleName',
        'accessibilityAttributes',
    ]),
    __metadata("design:paramtypes", [])
], AvatarComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: AvatarComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-avatar',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'interactive',
                        'icon',
                        'fallbackIcon',
                        'initials',
                        'shape',
                        'size',
                        'colorScheme',
                        'accessibleName',
                        'accessibilityAttributes',
                    ],
                    exportAs: 'ui5Avatar',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], interactive: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { AvatarComponent };
