// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-suggestion-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/SuggestionItem.js';

let SuggestionItemComponent = class SuggestionItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SuggestionItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: SuggestionItemComponent, isStandalone: true, selector: "ui5-suggestion-item", inputs: { text: "text", additionalText: "additionalText" }, exportAs: ["ui5SuggestionItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SuggestionItemComponent = __decorate([
    ProxyInputs(['text', 'additionalText']),
    __metadata("design:paramtypes", [])
], SuggestionItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SuggestionItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-suggestion-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'additionalText'],
                    exportAs: 'ui5SuggestionItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { SuggestionItemComponent };
