// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-popover.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Popover.js';

let PopoverComponent = class PopoverComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PopoverComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: PopoverComponent, isStandalone: true, selector: "ui5-popover", inputs: { initialFocus: "initialFocus", preventFocusRestore: ["preventFocusRestore", "preventFocusRestore", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibleRole: "accessibleRole", preventInitialFocus: ["preventInitialFocus", "preventInitialFocus", booleanAttribute], open: ["open", "open", booleanAttribute], headerText: "headerText", placement: "placement", horizontalAlign: "horizontalAlign", verticalAlign: "verticalAlign", modal: ["modal", "modal", booleanAttribute], hideArrow: ["hideArrow", "hideArrow", booleanAttribute], allowTargetOverlap: ["allowTargetOverlap", "allowTargetOverlap", booleanAttribute], opener: "opener" }, outputs: { ui5BeforeOpen: "ui5BeforeOpen", ui5Open: "ui5Open", ui5BeforeClose: "ui5BeforeClose", ui5Close: "ui5Close" }, exportAs: ["ui5Popover"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
PopoverComponent = __decorate([
    ProxyInputs([
        'initialFocus',
        'preventFocusRestore',
        'accessibleName',
        'accessibleNameRef',
        'accessibleRole',
        'preventInitialFocus',
        'open',
        'headerText',
        'placement',
        'horizontalAlign',
        'verticalAlign',
        'modal',
        'hideArrow',
        'allowTargetOverlap',
        'opener',
    ]),
    ProxyOutputs([
        'before-open: ui5BeforeOpen',
        'open: ui5Open',
        'before-close: ui5BeforeClose',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], PopoverComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: PopoverComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-popover',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'initialFocus',
                        'preventFocusRestore',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibleRole',
                        'preventInitialFocus',
                        'open',
                        'headerText',
                        'placement',
                        'horizontalAlign',
                        'verticalAlign',
                        'modal',
                        'hideArrow',
                        'allowTargetOverlap',
                        'opener',
                    ],
                    outputs: ['ui5BeforeOpen', 'ui5Open', 'ui5BeforeClose', 'ui5Close'],
                    exportAs: 'ui5Popover',
                }]
        }], ctorParameters: () => [], propDecorators: { preventFocusRestore: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], preventInitialFocus: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], modal: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hideArrow: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], allowTargetOverlap: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { PopoverComponent };
