// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-time-picker.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TimePicker.js';

let TimePickerComponent = class TimePickerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimePickerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TimePickerComponent, isStandalone: true, selector: "ui5-time-picker", inputs: { value: "value", name: "name", valueState: "valueState", disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], placeholder: "placeholder", formatPattern: "formatPattern", open: ["open", "open", booleanAttribute] }, outputs: { ui5Change: "ui5Change", ui5Input: "ui5Input", ui5Open: "ui5Open", ui5Close: "ui5Close" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);", "input": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5TimePicker"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TimePickerComponent = __decorate([
    ProxyInputs([
        'value',
        'name',
        'valueState',
        'disabled',
        'readonly',
        'placeholder',
        'formatPattern',
        'open',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'input: ui5Input',
        'open: ui5Open',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], TimePickerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TimePickerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-time-picker',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'value',
                        'name',
                        'valueState',
                        'disabled',
                        'readonly',
                        'placeholder',
                        'formatPattern',
                        'open',
                    ],
                    outputs: ['ui5Change', 'ui5Input', 'ui5Open', 'ui5Close'],
                    exportAs: 'ui5TimePicker',
                    hostDirectives: [GenericControlValueAccessor],
                    host: {
                        '(change)': '_cva?.onChange?.(cvaValue);',
                        '(input)': '_cva?.onChange?.(cvaValue);',
                    },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], open: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TimePickerComponent };
