// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-table-growing.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TableGrowing.js';

let TableGrowingComponent = class TableGrowingComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableGrowingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TableGrowingComponent, isStandalone: true, selector: "ui5-table-growing", inputs: { type: "type", growingText: "growingText", growingSubText: "growingSubText" }, outputs: { ui5LoadMore: "ui5LoadMore" }, exportAs: ["ui5TableGrowing"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableGrowingComponent = __decorate([
    ProxyInputs(['type', 'growingText', 'growingSubText']),
    ProxyOutputs(['load-more: ui5LoadMore']),
    __metadata("design:paramtypes", [])
], TableGrowingComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableGrowingComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table-growing',
                    template: '<ng-content></ng-content>',
                    inputs: ['type', 'growingText', 'growingSubText'],
                    outputs: ['ui5LoadMore'],
                    exportAs: 'ui5TableGrowing',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { TableGrowingComponent };
