// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-tab.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Tab.js';

let TabComponent = class TabComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TabComponent, isStandalone: true, selector: "ui5-tab", inputs: { text: "text", disabled: ["disabled", "disabled", booleanAttribute], additionalText: "additionalText", icon: "icon", design: "design", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5Tab"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TabComponent = __decorate([
    ProxyInputs([
        'text',
        'disabled',
        'additionalText',
        'icon',
        'design',
        'selected',
    ]),
    __metadata("design:paramtypes", [])
], TabComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TabComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-tab',
                    template: '<ng-content></ng-content>',
                    inputs: ['text', 'disabled', 'additionalText', 'icon', 'design', 'selected'],
                    exportAs: 'ui5Tab',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TabComponent };
