// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-message-strip.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/MessageStrip.js';

let MessageStripComponent = class MessageStripComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MessageStripComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MessageStripComponent, isStandalone: true, selector: "ui5-message-strip", inputs: { design: "design", colorScheme: "colorScheme", hideIcon: ["hideIcon", "hideIcon", booleanAttribute], hideCloseButton: ["hideCloseButton", "hideCloseButton", booleanAttribute] }, outputs: { ui5Close: "ui5Close" }, exportAs: ["ui5MessageStrip"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MessageStripComponent = __decorate([
    ProxyInputs(['design', 'colorScheme', 'hideIcon', 'hideCloseButton']),
    ProxyOutputs(['close: ui5Close']),
    __metadata("design:paramtypes", [])
], MessageStripComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MessageStripComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-message-strip',
                    template: '<ng-content></ng-content>',
                    inputs: ['design', 'colorScheme', 'hideIcon', 'hideCloseButton'],
                    outputs: ['ui5Close'],
                    exportAs: 'ui5MessageStrip',
                }]
        }], ctorParameters: () => [], propDecorators: { hideIcon: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hideCloseButton: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { MessageStripComponent };
