// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-shell-bar.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/ShellBar.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let ShellBarComponent = class ShellBarComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ShellBarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ShellBarComponent, isStandalone: true, selector: "ui5-shellbar", inputs: { primaryTitle: "primaryTitle", secondaryTitle: "secondaryTitle", notificationsCount: "notificationsCount", showNotifications: ["showNotifications", "showNotifications", booleanAttribute], showProductSwitch: ["showProductSwitch", "showProductSwitch", booleanAttribute], showSearchField: ["showSearchField", "showSearchField", booleanAttribute], accessibilityAttributes: "accessibilityAttributes" }, outputs: { ui5NotificationsClick: "ui5NotificationsClick", ui5ProfileClick: "ui5ProfileClick", ui5ProductSwitchClick: "ui5ProductSwitchClick", ui5LogoClick: "ui5LogoClick", ui5MenuItemClick: "ui5MenuItemClick", ui5SearchButtonClick: "ui5SearchButtonClick" }, exportAs: ["ui5Shellbar"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ShellBarComponent = __decorate([
    ProxyInputs([
        'primaryTitle',
        'secondaryTitle',
        'notificationsCount',
        'showNotifications',
        'showProductSwitch',
        'showSearchField',
        'accessibilityAttributes',
    ]),
    ProxyOutputs([
        'notifications-click: ui5NotificationsClick',
        'profile-click: ui5ProfileClick',
        'product-switch-click: ui5ProductSwitchClick',
        'logo-click: ui5LogoClick',
        'menu-item-click: ui5MenuItemClick',
        'search-button-click: ui5SearchButtonClick',
    ]),
    __metadata("design:paramtypes", [])
], ShellBarComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ShellBarComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-shellbar',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'primaryTitle',
                        'secondaryTitle',
                        'notificationsCount',
                        'showNotifications',
                        'showProductSwitch',
                        'showSearchField',
                        'accessibilityAttributes',
                    ],
                    outputs: [
                        'ui5NotificationsClick',
                        'ui5ProfileClick',
                        'ui5ProductSwitchClick',
                        'ui5LogoClick',
                        'ui5MenuItemClick',
                        'ui5SearchButtonClick',
                    ],
                    exportAs: 'ui5Shellbar',
                }]
        }], ctorParameters: () => [], propDecorators: { showNotifications: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], showProductSwitch: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], showSearchField: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ShellBarComponent };
