// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-list.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/List.js';

let ListComponent = class ListComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ListComponent, isStandalone: true, selector: "ui5-list", inputs: { headerText: "headerText", footerText: "footerText", indent: ["indent", "indent", booleanAttribute], selectionMode: "selectionMode", noDataText: "noDataText", separators: "separators", growing: "growing", growingButtonText: "growingButtonText", loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", accessibleRole: "accessibleRole" }, outputs: { ui5ItemClick: "ui5ItemClick", ui5ItemClose: "ui5ItemClose", ui5ItemToggle: "ui5ItemToggle", ui5ItemDelete: "ui5ItemDelete", ui5SelectionChange: "ui5SelectionChange", ui5LoadMore: "ui5LoadMore", ui5MoveOver: "ui5MoveOver", ui5Move: "ui5Move" }, exportAs: ["ui5List"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ListComponent = __decorate([
    ProxyInputs([
        'headerText',
        'footerText',
        'indent',
        'selectionMode',
        'noDataText',
        'separators',
        'growing',
        'growingButtonText',
        'loading',
        'loadingDelay',
        'accessibleName',
        'accessibleNameRef',
        'accessibleRole',
    ]),
    ProxyOutputs([
        'item-click: ui5ItemClick',
        'item-close: ui5ItemClose',
        'item-toggle: ui5ItemToggle',
        'item-delete: ui5ItemDelete',
        'selection-change: ui5SelectionChange',
        'load-more: ui5LoadMore',
        'move-over: ui5MoveOver',
        'move: ui5Move',
    ]),
    __metadata("design:paramtypes", [])
], ListComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ListComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-list',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'headerText',
                        'footerText',
                        'indent',
                        'selectionMode',
                        'noDataText',
                        'separators',
                        'growing',
                        'growingButtonText',
                        'loading',
                        'loadingDelay',
                        'accessibleName',
                        'accessibleNameRef',
                        'accessibleRole',
                    ],
                    outputs: [
                        'ui5ItemClick',
                        'ui5ItemClose',
                        'ui5ItemToggle',
                        'ui5ItemDelete',
                        'ui5SelectionChange',
                        'ui5LoadMore',
                        'ui5MoveOver',
                        'ui5Move',
                    ],
                    exportAs: 'ui5List',
                }]
        }], ctorParameters: () => [], propDecorators: { indent: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], loading: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ListComponent };
