// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-combo-box-item-group.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ComboBoxItemGroup.js';

let ComboBoxItemGroupComponent = class ComboBoxItemGroupComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ComboBoxItemGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ComboBoxItemGroupComponent, isStandalone: true, selector: "ui5-cb-item-group", inputs: { headerText: "headerText" }, exportAs: ["ui5CbItemGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ComboBoxItemGroupComponent = __decorate([
    ProxyInputs(['headerText']),
    __metadata("design:paramtypes", [])
], ComboBoxItemGroupComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ComboBoxItemGroupComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-cb-item-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['headerText'],
                    exportAs: 'ui5CbItemGroup',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ComboBoxItemGroupComponent };
