// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-select.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Select.js';

let SelectComponent = class SelectComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.value = val;
        this.cdr.detectChanges();
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SelectComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: SelectComponent, isStandalone: true, selector: "ui5-select", inputs: { disabled: ["disabled", "disabled", booleanAttribute], name: "name", valueState: "valueState", required: ["required", "required", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", value: "value" }, outputs: { ui5Change: "ui5Change", ui5LiveChange: "ui5LiveChange", ui5Open: "ui5Open", ui5Close: "ui5Close" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5Select"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
SelectComponent = __decorate([
    ProxyInputs([
        'disabled',
        'name',
        'valueState',
        'required',
        'readonly',
        'accessibleName',
        'accessibleNameRef',
        'value',
    ]),
    ProxyOutputs([
        'change: ui5Change',
        'live-change: ui5LiveChange',
        'open: ui5Open',
        'close: ui5Close',
    ]),
    __metadata("design:paramtypes", [])
], SelectComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: SelectComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-select',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'name',
                        'valueState',
                        'required',
                        'readonly',
                        'accessibleName',
                        'accessibleNameRef',
                        'value',
                    ],
                    outputs: ['ui5Change', 'ui5LiveChange', 'ui5Open', 'ui5Close'],
                    exportAs: 'ui5Select',
                    hostDirectives: [GenericControlValueAccessor],
                    host: { '(change)': '_cva?.onChange?.(cvaValue);' },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { SelectComponent };
