// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-illustrated-message.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/IllustratedMessage.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';

let IllustratedMessageComponent = class IllustratedMessageComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: IllustratedMessageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: IllustratedMessageComponent, isStandalone: true, selector: "ui5-illustrated-message", inputs: { name: "name", design: "design", subtitleText: "subtitleText", titleText: "titleText", accessibleNameRef: "accessibleNameRef" }, exportAs: ["ui5IllustratedMessage"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
IllustratedMessageComponent = __decorate([
    ProxyInputs([
        'name',
        'design',
        'subtitleText',
        'titleText',
        'accessibleNameRef',
    ]),
    __metadata("design:paramtypes", [])
], IllustratedMessageComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: IllustratedMessageComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-illustrated-message',
                    template: '<ng-content></ng-content>',
                    inputs: ['name', 'design', 'subtitleText', 'titleText', 'accessibleNameRef'],
                    exportAs: 'ui5IllustratedMessage',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { IllustratedMessageComponent };
