// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-carousel.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Carousel.js';

let CarouselComponent = class CarouselComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CarouselComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: CarouselComponent, isStandalone: true, selector: "ui5-carousel", inputs: { accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", cyclic: ["cyclic", "cyclic", booleanAttribute], itemsPerPage: "itemsPerPage", hideNavigationArrows: ["hideNavigationArrows", "hideNavigationArrows", booleanAttribute], hidePageIndicator: ["hidePageIndicator", "hidePageIndicator", booleanAttribute], pageIndicatorType: "pageIndicatorType", backgroundDesign: "backgroundDesign", pageIndicatorBackgroundDesign: "pageIndicatorBackgroundDesign", pageIndicatorBorderDesign: "pageIndicatorBorderDesign", arrowsPlacement: "arrowsPlacement" }, outputs: { ui5Navigate: "ui5Navigate" }, exportAs: ["ui5Carousel"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CarouselComponent = __decorate([
    ProxyInputs([
        'accessibleName',
        'accessibleNameRef',
        'cyclic',
        'itemsPerPage',
        'hideNavigationArrows',
        'hidePageIndicator',
        'pageIndicatorType',
        'backgroundDesign',
        'pageIndicatorBackgroundDesign',
        'pageIndicatorBorderDesign',
        'arrowsPlacement',
    ]),
    ProxyOutputs(['navigate: ui5Navigate']),
    __metadata("design:paramtypes", [])
], CarouselComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CarouselComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-carousel',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accessibleName',
                        'accessibleNameRef',
                        'cyclic',
                        'itemsPerPage',
                        'hideNavigationArrows',
                        'hidePageIndicator',
                        'pageIndicatorType',
                        'backgroundDesign',
                        'pageIndicatorBackgroundDesign',
                        'pageIndicatorBorderDesign',
                        'arrowsPlacement',
                    ],
                    outputs: ['ui5Navigate'],
                    exportAs: 'ui5Carousel',
                }]
        }], ctorParameters: () => [], propDecorators: { cyclic: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hideNavigationArrows: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], hidePageIndicator: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { CarouselComponent };
