// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-product-switch.mjs
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/ProductSwitch.js';

class ProductSwitchComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProductSwitchComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ProductSwitchComponent, isStandalone: true, selector: "ui5-product-switch", exportAs: ["ui5ProductSwitch"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProductSwitchComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-product-switch',
                    template: '<ng-content></ng-content>',
                    exportAs: 'ui5ProductSwitch',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ProductSwitchComponent };
