// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-form.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Form.js';

let FormComponent = class FormComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: FormComponent, isStandalone: true, selector: "ui5-form", inputs: { layout: "layout", labelSpan: "labelSpan", headerText: "headerText", itemSpacing: "itemSpacing" }, exportAs: ["ui5Form"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
FormComponent = __decorate([
    ProxyInputs(['layout', 'labelSpan', 'headerText', 'itemSpacing']),
    __metadata("design:paramtypes", [])
], FormComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: FormComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-form',
                    template: '<ng-content></ng-content>',
                    inputs: ['layout', 'labelSpan', 'headerText', 'itemSpacing'],
                    exportAs: 'ui5Form',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { FormComponent };
