// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-title.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Title.js';

let TitleComponent = class TitleComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: TitleComponent, isStandalone: true, selector: "ui5-title", inputs: { wrappingType: "wrappingType", level: "level" }, exportAs: ["ui5Title"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TitleComponent = __decorate([
    ProxyInputs(['wrappingType', 'level']),
    __metadata("design:paramtypes", [])
], TitleComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TitleComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-title',
                    template: '<ng-content></ng-content>',
                    inputs: ['wrappingType', 'level'],
                    exportAs: 'ui5Title',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { TitleComponent };
