// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-table-row.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/TableRow.js';

let TableRowComponent = class TableRowComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableRowComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: TableRowComponent, isStandalone: true, selector: "ui5-table-row", inputs: { rowKey: "rowKey", interactive: ["interactive", "interactive", booleanAttribute], navigated: ["navigated", "navigated", booleanAttribute] }, exportAs: ["ui5TableRow"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
TableRowComponent = __decorate([
    ProxyInputs(['rowKey', 'interactive', 'navigated']),
    __metadata("design:paramtypes", [])
], TableRowComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: TableRowComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-table-row',
                    template: '<ng-content></ng-content>',
                    inputs: ['rowKey', 'interactive', 'navigated'],
                    exportAs: 'ui5TableRow',
                }]
        }], ctorParameters: () => [], propDecorators: { interactive: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], navigated: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { TableRowComponent };
