// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-toolbar.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Toolbar.js';

let ToolbarComponent = class ToolbarComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ToolbarComponent, isStandalone: true, selector: "ui5-toolbar", inputs: { alignContent: "alignContent", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef", design: "design" }, exportAs: ["ui5Toolbar"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarComponent = __decorate([
    ProxyInputs(['alignContent', 'accessibleName', 'accessibleNameRef', 'design']),
    __metadata("design:paramtypes", [])
], ToolbarComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar',
                    template: '<ng-content></ng-content>',
                    inputs: ['alignContent', 'accessibleName', 'accessibleNameRef', 'design'],
                    exportAs: 'ui5Toolbar',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ToolbarComponent };
