// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-notification-list-group-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/NotificationListGroupItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let NotificationListGroupItemComponent = class NotificationListGroupItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListGroupItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: NotificationListGroupItemComponent, isStandalone: true, selector: "ui5-li-notification-group", inputs: { titleText: "titleText", read: ["read", "read", booleanAttribute], loading: ["loading", "loading", booleanAttribute], loadingDelay: "loadingDelay", collapsed: ["collapsed", "collapsed", booleanAttribute] }, outputs: { ui5Toggle: "ui5Toggle" }, exportAs: ["ui5LiNotificationGroup"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
NotificationListGroupItemComponent = __decorate([
    ProxyInputs(['titleText', 'read', 'loading', 'loadingDelay', 'collapsed']),
    ProxyOutputs(['toggle: ui5Toggle']),
    __metadata("design:paramtypes", [])
], NotificationListGroupItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: NotificationListGroupItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-li-notification-group',
                    template: '<ng-content></ng-content>',
                    inputs: ['titleText', 'read', 'loading', 'loadingDelay', 'collapsed'],
                    outputs: ['ui5Toggle'],
                    exportAs: 'ui5LiNotificationGroup',
                }]
        }], ctorParameters: () => [], propDecorators: { read: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], loading: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], collapsed: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { NotificationListGroupItemComponent };
