// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-step-input.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/StepInput.js';

let StepInputComponent = class StepInputComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: StepInputComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: StepInputComponent, isStandalone: true, selector: "ui5-step-input", inputs: { value: "value", min: "min", max: "max", step: "step", valueState: "valueState", required: ["required", "required", booleanAttribute], disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], placeholder: "placeholder", name: "name", valuePrecision: "valuePrecision", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5Change: "ui5Change", ui5ValueStateChange: "ui5ValueStateChange" }, exportAs: ["ui5StepInput"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
StepInputComponent = __decorate([
    ProxyInputs([
        'value',
        'min',
        'max',
        'step',
        'valueState',
        'required',
        'disabled',
        'readonly',
        'placeholder',
        'name',
        'valuePrecision',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs(['change: ui5Change', 'value-state-change: ui5ValueStateChange']),
    __metadata("design:paramtypes", [])
], StepInputComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: StepInputComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-step-input',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'value',
                        'min',
                        'max',
                        'step',
                        'valueState',
                        'required',
                        'disabled',
                        'readonly',
                        'placeholder',
                        'name',
                        'valuePrecision',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: ['ui5Change', 'ui5ValueStateChange'],
                    exportAs: 'ui5StepInput',
                }]
        }], ctorParameters: () => [], propDecorators: { required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { StepInputComponent };
