// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-wizard-step.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/WizardStep.js';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';

let WizardStepComponent = class WizardStepComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WizardStepComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: WizardStepComponent, isStandalone: true, selector: "ui5-wizard-step", inputs: { titleText: "titleText", subtitleText: "subtitleText", icon: "icon", disabled: ["disabled", "disabled", booleanAttribute], selected: ["selected", "selected", booleanAttribute], branching: ["branching", "branching", booleanAttribute] }, exportAs: ["ui5WizardStep"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
WizardStepComponent = __decorate([
    ProxyInputs([
        'titleText',
        'subtitleText',
        'icon',
        'disabled',
        'selected',
        'branching',
    ]),
    __metadata("design:paramtypes", [])
], WizardStepComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: WizardStepComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-wizard-step',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'titleText',
                        'subtitleText',
                        'icon',
                        'disabled',
                        'selected',
                        'branching',
                    ],
                    exportAs: 'ui5WizardStep',
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], branching: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { WizardStepComponent };
