// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-option.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Option.js';

let OptionComponent = class OptionComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: OptionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: OptionComponent, isStandalone: true, selector: "ui5-option", inputs: { value: "value", icon: "icon", additionalText: "additionalText", tooltip: "tooltip", selected: ["selected", "selected", booleanAttribute] }, exportAs: ["ui5Option"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
OptionComponent = __decorate([
    ProxyInputs(['value', 'icon', 'additionalText', 'tooltip', 'selected']),
    __metadata("design:paramtypes", [])
], OptionComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: OptionComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-option',
                    template: '<ng-content></ng-content>',
                    inputs: ['value', 'icon', 'additionalText', 'tooltip', 'selected'],
                    exportAs: 'ui5Option',
                }]
        }], ctorParameters: () => [], propDecorators: { selected: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { OptionComponent };
