// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-breadcrumbs-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/BreadcrumbsItem.js';

let BreadcrumbsItemComponent = class BreadcrumbsItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BreadcrumbsItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: BreadcrumbsItemComponent, isStandalone: true, selector: "ui5-breadcrumbs-item", inputs: { href: "href", target: "target", accessibleName: "accessibleName" }, exportAs: ["ui5BreadcrumbsItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
BreadcrumbsItemComponent = __decorate([
    ProxyInputs(['href', 'target', 'accessibleName']),
    __metadata("design:paramtypes", [])
], BreadcrumbsItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: BreadcrumbsItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-breadcrumbs-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['href', 'target', 'accessibleName'],
                    exportAs: 'ui5BreadcrumbsItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { BreadcrumbsItemComponent };
