// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-color-picker.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ColorPicker.js';

let ColorPickerComponent = class ColorPickerComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPickerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ColorPickerComponent, isStandalone: true, selector: "ui5-color-picker", inputs: { value: "value", name: "name" }, outputs: { ui5Change: "ui5Change" }, exportAs: ["ui5ColorPicker"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ColorPickerComponent = __decorate([
    ProxyInputs(['value', 'name']),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], ColorPickerComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ColorPickerComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-color-picker',
                    template: '<ng-content></ng-content>',
                    inputs: ['value', 'name'],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5ColorPicker',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ColorPickerComponent };
