// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-i18n.mjs
import * as i0 from '@angular/core';
import { InjectionToken, inject, Injectable, Inject, Pipe, NgModule } from '@angular/core';
import { registerI18nLoader, getI18nBundle } from '@ui5/webcomponents-base/dist/i18nBundle.js';
import parseProperties from '@ui5/webcomponents-base/dist/PropertiesFileFormat.js';
import { getLanguage, setLanguage, setFetchDefaultLanguage } from '@ui5/webcomponents-base/dist/config/Language.js';
import { BehaviorSubject, isObservable, firstValueFrom, from, filter, first, tap, switchMap, map, of, Subject, distinctUntilChanged, takeUntil } from 'rxjs';

const I18N_ROOT_CONFIG = new InjectionToken('rootI18nConfig');
const I18N_NAMESPACE = new InjectionToken('i18nNamespace');
const I18N_TRANSLATIONS = new InjectionToken('i18nTranslations');

class I18nService {
    constructor(config, namespace, bundles = {}) {
        this.config = config;
        this.namespace = namespace;
        this.bundles = bundles;
        this.parent = inject(I18nService, { skipSelf: true, optional: true });
        this.currentLanguage$ = new BehaviorSubject(getLanguage() || this.config.language);
        this.loadedLanguages$ = new BehaviorSubject([]);
        Object.keys(bundles).forEach((lang) => {
            registerI18nLoader(this.namespace, lang, async (localeId) => {
                const val = this.bundles[localeId];
                const result = isObservable(val) ? await firstValueFrom(val) : await val;
                setTimeout(() => {
                    this.loadedLanguages$.next([
                        ...new Set([...this.loadedLanguages$.value, localeId]).values()
                    ]);
                });
                if (typeof result === 'string') {
                    return parseProperties(result);
                }
                return result;
            });
        });
        this.i18nBundle$ = from(getI18nBundle(this.namespace));
    }
    currentLanguage() {
        if (this.parent) {
            return this.parent.currentLanguage();
        }
        return this.currentLanguage$.value;
    }
    currentLanguageEvent() {
        if (this.parent) {
            return this.parent.currentLanguageEvent();
        }
        return this.currentLanguage$;
    }
    setLanguage(language) {
        if (this.parent) {
            return this.parent.setLanguage(language);
        }
        setLanguage(language);
        this.loadedLanguages$.pipe(filter(langs => langs.includes(language)), first(), tap(() => this.currentLanguage$.next(language))).subscribe();
    }
    getText(key, ...placeholders) {
        return this.currentLanguageEvent().pipe(switchMap(() => this.i18nBundle$), map((bundle) => bundle.getText({ key, defaultText: key }, ...placeholders)), switchMap((text) => {
            if (text === key && this.parent) {
                return this.parent.getText(key, ...placeholders);
            }
            return of(text);
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: I18nService, deps: [{ token: I18N_ROOT_CONFIG }, { token: I18N_NAMESPACE }, { token: I18N_TRANSLATIONS }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: I18nService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: I18nService, decorators: [{
            type: Injectable
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [I18N_ROOT_CONFIG]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [I18N_NAMESPACE]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [I18N_TRANSLATIONS]
                }] }] });

class I18nPipe {
    constructor(changeDetectorRef, i18nService) {
        this.changeDetectorRef = changeDetectorRef;
        this.i18nService = i18nService;
        this._lastTranslationValue = '';
        this._destroyed$ = new Subject();
    }
    transform(value, ...params) {
        const paramsChanged = !this._lastValue || params.some((param, index) => param !== this._lastValueParams[index]);
        if (value === this._lastValue && !paramsChanged) {
            return this._lastTranslationValue;
        }
        this._lastValue = value;
        this._lastValueParams = params;
        if (this._subscription) {
            this._subscription.unsubscribe();
            this._subscription = undefined;
        }
        this._subscription = this.i18nService.getText(value, ...params)
            .pipe(distinctUntilChanged(), takeUntil(this._destroyed$)).subscribe((text) => {
            this._lastTranslationValue = text;
            this.changeDetectorRef.markForCheck();
        });
        return this._lastTranslationValue;
    }
    ngOnDestroy() {
        this._destroyed$.next();
        this._destroyed$.complete();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: I18nPipe, deps: [{ token: i0.ChangeDetectorRef }, { token: I18nService }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: I18nPipe, isStandalone: true, name: "ui5I18n", pure: false }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: I18nPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'ui5I18n',
                    pure: false,
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.ChangeDetectorRef }, { type: I18nService }] });

function resolveTranslationsProvider(translations) {
    if (!translations) {
        return {
            provide: I18N_TRANSLATIONS,
            useValue: {}
        };
    }
    if (typeof translations.useFactory === 'function') {
        return {
            provide: I18N_TRANSLATIONS,
            useFactory: translations.useFactory,
            deps: translations.deps || [],
        };
    }
    return {
        provide: I18N_TRANSLATIONS,
        useValue: translations
    };
}

let childIds = 0;
function i18nRootProviders(config) {
    return [
        {
            provide: I18N_ROOT_CONFIG,
            useFactory: () => {
                if (config.language) {
                    setLanguage(config.language);
                }
                if (config.fetchDefaultLanguage) {
                    setFetchDefaultLanguage(config.fetchDefaultLanguage);
                }
                return config;
            }
        },
        {
            provide: I18N_NAMESPACE,
            useValue: config.bundle?.name || 'root_i18n'
        },
        resolveTranslationsProvider(config.bundle?.translations),
        I18nService
    ];
}
function i18nChildProviders(config) {
    return [
        {
            provide: I18N_NAMESPACE,
            useValue: config?.name || ++childIds + '_i18n'
        },
        resolveTranslationsProvider(config?.translations),
        I18nService
    ];
}

class Ui5I18nModule {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    constructor(config) {
        this.config = config;
    }
    static forRoot(config = {}) {
        return {
            ngModule: Ui5I18nModule,
            providers: [
                i18nRootProviders(config)
            ]
        };
    }
    static forChild(config = {}) {
        return {
            ngModule: Ui5I18nModule,
            providers: [
                i18nChildProviders(config)
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5I18nModule, deps: [{ token: I18N_ROOT_CONFIG }], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.0.9", ngImport: i0, type: Ui5I18nModule, imports: [I18nPipe], exports: [I18nPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5I18nModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: Ui5I18nModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [I18nPipe],
                    exports: [I18nPipe]
                }]
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [I18N_ROOT_CONFIG]
                }] }] });

/**
 * Generated bundle index. Do not edit.
 */

export { I18nPipe, I18nService, Ui5I18nModule, i18nChildProviders, i18nRootProviders };
