// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-calendar-date.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/CalendarDate.js';

let CalendarDateComponent = class CalendarDateComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarDateComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: CalendarDateComponent, isStandalone: true, selector: "ui5-date", inputs: { value: "value" }, exportAs: ["ui5Date"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CalendarDateComponent = __decorate([
    ProxyInputs(['value']),
    __metadata("design:paramtypes", [])
], CalendarDateComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarDateComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-date',
                    template: '<ng-content></ng-content>',
                    inputs: ['value'],
                    exportAs: 'ui5Date',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { CalendarDateComponent };
