// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-fiori-shell-bar-item.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, Component } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/ShellBarItem.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';

let ShellBarItemComponent = class ShellBarItemComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ShellBarItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.9", type: ShellBarItemComponent, isStandalone: true, selector: "ui5-shellbar-item", inputs: { icon: "icon", text: "text", count: "count" }, outputs: { ui5Click: "ui5Click" }, exportAs: ["ui5ShellbarItem"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ShellBarItemComponent = __decorate([
    ProxyInputs(['icon', 'text', 'count']),
    ProxyOutputs(['click: ui5Click']),
    __metadata("design:paramtypes", [])
], ShellBarItemComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ShellBarItemComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-shellbar-item',
                    template: '<ng-content></ng-content>',
                    inputs: ['icon', 'text', 'count'],
                    outputs: ['ui5Click'],
                    exportAs: 'ui5ShellbarItem',
                }]
        }], ctorParameters: () => [] });

/**
 * Generated bundle index. Do not edit.
 */

export { ShellBarItemComponent };
