// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-calendar.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/Calendar.js';

let CalendarComponent = class CalendarComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: CalendarComponent, isStandalone: true, selector: "ui5-calendar", inputs: { primaryCalendarType: "primaryCalendarType", secondaryCalendarType: "secondaryCalendarType", formatPattern: "formatPattern", minDate: "minDate", maxDate: "maxDate", selectionMode: "selectionMode", hideWeekNumbers: ["hideWeekNumbers", "hideWeekNumbers", booleanAttribute] }, outputs: { ui5SelectionChange: "ui5SelectionChange" }, exportAs: ["ui5Calendar"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
CalendarComponent = __decorate([
    ProxyInputs([
        'primaryCalendarType',
        'secondaryCalendarType',
        'formatPattern',
        'minDate',
        'maxDate',
        'selectionMode',
        'hideWeekNumbers',
    ]),
    ProxyOutputs(['selection-change: ui5SelectionChange']),
    __metadata("design:paramtypes", [])
], CalendarComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: CalendarComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-calendar',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'primaryCalendarType',
                        'secondaryCalendarType',
                        'formatPattern',
                        'minDate',
                        'maxDate',
                        'selectionMode',
                        'hideWeekNumbers',
                    ],
                    outputs: ['ui5SelectionChange'],
                    exportAs: 'ui5Calendar',
                }]
        }], ctorParameters: () => [], propDecorators: { hideWeekNumbers: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { CalendarComponent };
