// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-progress-indicator.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ProgressIndicator.js';

let ProgressIndicatorComponent = class ProgressIndicatorComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProgressIndicatorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ProgressIndicatorComponent, isStandalone: true, selector: "ui5-progress-indicator", inputs: { accessibleName: "accessibleName", hideValue: ["hideValue", "hideValue", booleanAttribute], value: "value", displayValue: "displayValue", valueState: "valueState" }, exportAs: ["ui5ProgressIndicator"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ProgressIndicatorComponent = __decorate([
    ProxyInputs([
        'accessibleName',
        'hideValue',
        'value',
        'displayValue',
        'valueState',
    ]),
    __metadata("design:paramtypes", [])
], ProgressIndicatorComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ProgressIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-progress-indicator',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'accessibleName',
                        'hideValue',
                        'value',
                        'displayValue',
                        'valueState',
                    ],
                    exportAs: 'ui5ProgressIndicator',
                }]
        }], ctorParameters: () => [], propDecorators: { hideValue: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ProgressIndicatorComponent };
