// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-toolbar-select.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/ToolbarSelect.js';

let ToolbarSelectComponent = class ToolbarSelectComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSelectComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: ToolbarSelectComponent, isStandalone: true, selector: "ui5-toolbar-select", inputs: { overflowPriority: "overflowPriority", preventOverflowClosing: ["preventOverflowClosing", "preventOverflowClosing", booleanAttribute], width: "width", valueState: "valueState", disabled: ["disabled", "disabled", booleanAttribute], accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5Change: "ui5Change", ui5Open: "ui5Open", ui5Close: "ui5Close" }, exportAs: ["ui5ToolbarSelect"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
ToolbarSelectComponent = __decorate([
    ProxyInputs([
        'overflowPriority',
        'preventOverflowClosing',
        'width',
        'valueState',
        'disabled',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs(['change: ui5Change', 'open: ui5Open', 'close: ui5Close']),
    __metadata("design:paramtypes", [])
], ToolbarSelectComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: ToolbarSelectComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-toolbar-select',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'overflowPriority',
                        'preventOverflowClosing',
                        'width',
                        'valueState',
                        'disabled',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: ['ui5Change', 'ui5Open', 'ui5Close'],
                    exportAs: 'ui5ToolbarSelect',
                }]
        }], ctorParameters: () => [], propDecorators: { preventOverflowClosing: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { ToolbarSelectComponent };
