// @ui5/webcomponents-ngx/fesm2022/ui5-webcomponents-ngx-main-radio-button.mjs
import { __metadata, __decorate } from 'tslib';
import * as i0 from '@angular/core';
import { ElementRef, inject, NgZone, ChangeDetectorRef, booleanAttribute, Component, Input } from '@angular/core';
import * as i1 from '@ui5/webcomponents-ngx/generic-cva';
import { GenericControlValueAccessor } from '@ui5/webcomponents-ngx/generic-cva';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import '@ui5/webcomponents/dist/RadioButton.js';

let RadioButtonComponent = class RadioButtonComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    set cvaValue(val) {
        this.element.checked = this.element.value === val;
    }
    get cvaValue() {
        return this.element.value;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this._cva = inject(GenericControlValueAccessor);
        this.cdr.detach();
        this._cva.host = this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: RadioButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: RadioButtonComponent, isStandalone: true, selector: "ui5-radio-button", inputs: { disabled: ["disabled", "disabled", booleanAttribute], readonly: ["readonly", "readonly", booleanAttribute], required: ["required", "required", booleanAttribute], checked: ["checked", "checked", booleanAttribute], text: "text", valueState: "valueState", name: "name", value: "value", wrappingType: "wrappingType", accessibleName: "accessibleName", accessibleNameRef: "accessibleNameRef" }, outputs: { ui5Change: "ui5Change" }, host: { listeners: { "change": "_cva?.onChange?.(cvaValue);" } }, exportAs: ["ui5RadioButton"], hostDirectives: [{ directive: i1.GenericControlValueAccessor }], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
RadioButtonComponent = __decorate([
    ProxyInputs([
        'disabled',
        'readonly',
        'required',
        'checked',
        'text',
        'valueState',
        'name',
        'value',
        'wrappingType',
        'accessibleName',
        'accessibleNameRef',
    ]),
    ProxyOutputs(['change: ui5Change']),
    __metadata("design:paramtypes", [])
], RadioButtonComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: RadioButtonComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-radio-button',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'disabled',
                        'readonly',
                        'required',
                        'checked',
                        'text',
                        'valueState',
                        'name',
                        'value',
                        'wrappingType',
                        'accessibleName',
                        'accessibleNameRef',
                    ],
                    outputs: ['ui5Change'],
                    exportAs: 'ui5RadioButton',
                    hostDirectives: [GenericControlValueAccessor],
                    host: { '(change)': '_cva?.onChange?.(cvaValue);' },
                }]
        }], ctorParameters: () => [], propDecorators: { disabled: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], readonly: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], required: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }], checked: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });

/**
 * Generated bundle index. Do not edit.
 */

export { RadioButtonComponent };
