// @ui5/webcomponents-ngx/esm2022/fiori/media-gallery/index.mjs
import { __decorate, __metadata } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, Input as InputDecorator, NgZone, booleanAttribute, inject, } from '@angular/core';
import '@ui5/webcomponents-fiori/dist/MediaGallery.js';
import { ProxyInputs, ProxyOutputs } from '@ui5/webcomponents-ngx/utils';
import * as i0 from "@angular/core";
let MediaGalleryComponent = class MediaGalleryComponent {
    get element() {
        return this.elementRef.nativeElement;
    }
    constructor() {
        this.elementRef = inject(ElementRef);
        this.zone = inject(NgZone);
        this.cdr = inject(ChangeDetectorRef);
        this.cdr.detach();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "16.1.0", version: "17.0.9", type: MediaGalleryComponent, isStandalone: true, selector: "ui5-media-gallery", inputs: { showAllThumbnails: ["showAllThumbnails", "showAllThumbnails", booleanAttribute], interactiveDisplayArea: ["interactiveDisplayArea", "interactiveDisplayArea", booleanAttribute], layout: "layout", menuHorizontalAlign: "menuHorizontalAlign", menuVerticalAlign: "menuVerticalAlign" }, outputs: { ui5SelectionChange: "ui5SelectionChange", ui5OverflowClick: "ui5OverflowClick", ui5DisplayAreaClick: "ui5DisplayAreaClick" }, exportAs: ["ui5MediaGallery"], ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
};
MediaGalleryComponent = __decorate([
    ProxyInputs([
        'showAllThumbnails',
        'interactiveDisplayArea',
        'layout',
        'menuHorizontalAlign',
        'menuVerticalAlign',
    ]),
    ProxyOutputs([
        'selection-change: ui5SelectionChange',
        'overflow-click: ui5OverflowClick',
        'display-area-click: ui5DisplayAreaClick',
    ]),
    __metadata("design:paramtypes", [])
], MediaGalleryComponent);
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.9", ngImport: i0, type: MediaGalleryComponent, decorators: [{
            type: Component,
            args: [{
                    standalone: true,
                    selector: 'ui5-media-gallery',
                    template: '<ng-content></ng-content>',
                    inputs: [
                        'showAllThumbnails',
                        'interactiveDisplayArea',
                        'layout',
                        'menuHorizontalAlign',
                        'menuVerticalAlign',
                    ],
                    outputs: ['ui5SelectionChange', 'ui5OverflowClick', 'ui5DisplayAreaClick'],
                    exportAs: 'ui5MediaGallery',
                }]
        }], ctorParameters: () => [], propDecorators: { showAllThumbnails: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }], interactiveDisplayArea: [{
                type: InputDecorator,
                args: [{ transform: booleanAttribute }]
            }] } });
export { MediaGalleryComponent };
